// @flowwwkr/design-system-tailwind@0.12.12/dist/tailwind/themes/index.js
const e = {
  auto: "auto",
  0: "0rem",
  "0.125": `${0.1}rem`,
  "0.25": `${0.2}rem`,
  "0.375": `${0.30000000000000004}rem`,
  "0.5": `${0.4}rem`,
  "0.75": `${0.6000000000000001}rem`,
  1: `${0.8}rem`,
  "1.25": "1rem",
  "1.5": `${1.2000000000000002}rem`,
  "1.75": `${1.4000000000000001}rem`,
  2: `${1.6}rem`,
  "2.25": `${1.8}rem`,
  "2.5": "2rem",
  "2.75": `${2.2}rem`,
  3: `${2.4000000000000004}rem`,
  "3.25": `${2.6}rem`,
  "3.5": `${2.8000000000000003}rem`,
  "3.75": "3rem",
  4: `${3.2}rem`,
  "4.25": `${3.4000000000000004}rem`,
  "4.5": `${3.6}rem`,
  "4.75": `${3.8000000000000003}rem`,
  5: "4rem",
  "5.5": `${4.4}rem`,
  6: `${4.800000000000001}rem`,
  "6.5": `${5.2}rem`,
  7: `${5.6000000000000005}rem`,
  "7.5": "6rem",
  8: `${6.4}rem`,
  "8.5": `${6.800000000000001}rem`,
  9: `${7.2}rem`,
  "9.5": `${7.6000000000000005}rem`,
  10: "8rem",
  11: `${8.8}rem`,
  12: `${9.600000000000001}rem`,
  13: `${10.4}rem`,
  14: `${11.200000000000001}rem`,
  15: "12rem",
  16: `${12.8}rem`,
  18: `${14.4}rem`,
  20: "16rem",
  22: `${17.6}rem`,
  24: `${19.200000000000003}rem`,
  26: `${20.8}rem`,
  28: `${22.400000000000002}rem`,
  30: "24rem",
  32: `${25.6}rem`,
  36: `${28.8}rem`,
  40: "32rem",
  44: `${35.2}rem`,
  48: `${38.400000000000006}rem`,
  52: `${41.6}rem`,
  56: `${44.800000000000004}rem`,
  60: "48rem",
  64: `${51.2}rem`,
  72: `${57.6}rem`,
  80: "64rem",
  88: `${70.4}rem`,
  96: `${76.80000000000001}rem`,
  104: `${83.2}rem`,
  112: `${89.60000000000001}rem`,
  120: "96rem",
  128: `${102.4}rem`
}, a = {
  "0%": "0%",
  "1%": "1%",
  "2%": "2%",
  "3%": "3%",
  "4%": "4%",
  "5%": "5%",
  "6%": "6%",
  "7%": "7%",
  "8%": "8%",
  "9%": "9%",
  "10%": "10%",
  "11%": "11%",
  "12%": "12%",
  "13%": "13%",
  "14%": "14%",
  "15%": "15%",
  "16%": "16%",
  "18%": "18%",
  "20%": "20%",
  "22%": "22%",
  "24%": "24%",
  "26%": "26%",
  "28%": "28%",
  "30%": "30%",
  "32%": "32%",
  "34%": "34%",
  "36%": "36%",
  "38%": "38%",
  "40%": "40%",
  "42%": "42%",
  "44%": "44%",
  "46%": "46%",
  "48%": "48%",
  "50%": "50%",
  "52%": "52%",
  "54%": "54%",
  "56%": "56%",
  "58%": "58%",
  "60%": "60%",
  "62%": "62%",
  "64%": "64%",
  "66%": "66%",
  "68%": "68%",
  "70%": "70%",
  "72%": "72%",
  "74%": "74%",
  "76%": "76%",
  "78%": "78%",
  "80%": "80%",
  "82%": "82%",
  "84%": "84%",
  "86%": "86%",
  "88%": "88%",
  "90%": "90%",
  "92%": "92%",
  "94%": "94%",
  "96%": "96%",
  "98%": "98%",
  "100%": "100%",
  "102%": "102%",
  "104%": "104%",
  "106%": "106%",
  "108%": "108%",
  "110%": "110%",
  "112%": "112%",
  "114%": "114%",
  "116%": "116%",
  "118%": "118%",
  "120%": "120%",
  "122%": "122%",
  "124%": "124%",
  "126%": "126%",
  "128%": "128%",
  "130%": "130%",
  "132%": "132%",
  "134%": "134%",
  "136%": "136%",
  "138%": "138%",
  "140%": "140%",
  "142%": "142%",
  "144%": "144%",
  "146%": "146%",
  "148%": "148%",
  "150%": "150%",
  "152%": "152%",
  "154%": "154%",
  "156%": "156%",
  "158%": "158%",
  "160%": "160%",
  "162%": "162%",
  "164%": "164%",
  "166%": "166%",
  "168%": "168%",
  "170%": "170%",
  "172%": "172%",
  "174%": "174%",
  "176%": "176%",
  "178%": "178%",
  "180%": "180%",
  "182%": "182%",
  "184%": "184%",
  "186%": "186%",
  "188%": "188%",
  "190%": "190%",
  "192%": "192%",
  "194%": "194%",
  "196%": "196%",
  "198%": "198%",
  "200%": "200%"
}, t = {
  ...e,
  ...a,
  auto: "auto",
  "xs-mobile": "32rem",
  "small-mobile": "48rem",
  mobile: "64rem",
  tablet: "76.8rem",
  laptop: "102.4rem",
  desktop: "128rem",
  "large-desktop": "144rem",
  "xl-desktop": "192rem",
  full: "100%",
  "min-content": "min-content",
  "max-content": "max-content",
  "fit-content": "fit-content",
  inherit: "inherit",
  initial: "initial",
  unset: "unset",
  screen: "screen"
}, n = {
  0: "0",
  1: "1",
  2: "2",
  3: "3",
  4: "4",
  5: "5",
  6: "6",
  7: "7",
  8: "8",
  9: "9",
  10: "10",
  11: "11",
  12: "12",
  13: "13",
  14: "14",
  15: "15",
  16: "16",
  17: "17",
  18: "18",
  19: "19",
  100: "100",
  101: "101",
  102: "102",
  103: "103",
  104: "104",
  105: "105",
  106: "106",
  107: "107",
  108: "108",
  109: "109",
  1001: "1001",
  1002: "1002",
  1003: "1003",
  1004: "1004",
  1005: "1005",
  1006: "1006",
  1007: "1007",
  1008: "1008",
  1009: "1009",
  1010: "1010",
  1101: "1101",
  1102: "1102",
  1103: "1103",
  1104: "1104",
  1105: "1105",
  1106: "1106",
  1107: "1107",
  1108: "1108",
  1109: "1109",
  1110: "1110"
}, m = {
  ...e,
  ...a
}, i = {
  solid: "solid",
  dashed: "dashed",
  dotted: "dotted",
  double: "double",
  groove: "groove",
  ridge: "ridge",
  hidden: "hidden",
  none: "none"
}, s = {
  0: `${0.8 * 0}rem`,
  "0.125": `${0.8 * 0.125}rem`,
  "0.25": `${0.8 * 0.25}rem`,
  "0.375": `${0.8 * 0.375}rem`,
  "0.5": `${0.8 * 0.5}rem`,
  "0.75": `${0.8 * 0.75}rem`,
  1: `${0.8 * 1}rem`
}, d = {
  0: `${0.8 * 0}rem`,
  "0.25": `${0.8 * 0.25}rem`,
  "0.5": `${0.8 * 0.5}rem`,
  "0.75": `${0.8 * 0.75}rem`,
  1: `${0.8 * 1}rem`,
  "1.5": `${0.8 * 1.5}rem`,
  2: `${0.8 * 2}rem`,
  "2.5": `${0.8 * 2.5}rem`,
  3: `${0.8 * 3}rem`,
  9999: `${0.8 * 9999}rem`,
  circle: "50%"
}, r = {
  ...e,
  ...a
}, c = {
  0: "0",
  "0.1": "0.1",
  "0.2": "0.2",
  "0.25": "0.25",
  "0.3": "0.3",
  "0.33": "0.33",
  "0.4": "0.4",
  "0.5": "0.5",
  "0.6": "0.6",
  "0.67": "0.67",
  "0.7": "0.7",
  "0.75": "0.75",
  "0.8": "0.8",
  "0.9": "0.9",
  1: "1"
}, g = {
  0: "0s",
  75: "75ms",
  100: "100ms",
  150: "150ms",
  200: "200ms",
  300: "300ms",
  500: "500ms",
  700: "700ms",
  1e3: "1000ms"
}, l = {
  0: "0",
  25: "0.25",
  50: "0.5",
  75: "0.75",
  90: "0.9",
  95: "0.95",
  100: "1",
  105: "1.05",
  110: "1.1",
  125: "1.25",
  150: "1.5"
}, u = {
  0: "0deg",
  1: "1deg",
  2: "2deg",
  3: "3deg",
  6: "6deg",
  12: "12deg",
  15: "15deg",
  30: "30deg",
  45: "45deg",
  60: "60deg",
  75: "75deg",
  90: "90deg",
  105: "105deg",
  120: "120deg",
  135: "135deg",
  150: "150deg",
  165: "165deg",
  180: "180deg"
}, p = {
  0: "0",
  1: "1",
  2: "2",
  3: "3",
  4: "4",
  5: "5",
  6: "6",
  7: "7",
  8: "8",
  9: "9",
  10: "10"
}, f = {
  1: "1",
  2: "2",
  3: "3",
  4: "4",
  5: "5",
  6: "6",
  7: "7",
  8: "8",
  9: "9",
  10: "10",
  11: "11",
  12: "12",
  13: "13",
  14: "14",
  15: "15",
  16: "16"
}, S = {
  1: "1",
  2: "2",
  3: "3",
  4: "4",
  5: "5",
  6: "6",
  7: "7",
  8: "8",
  9: "9",
  10: "10",
  11: "11",
  12: "12",
  13: "13",
  14: "14",
  15: "15",
  16: "16",
  none: "none",
  subgrid: "subgrid",
  "auto-fit": "auto-fit",
  "auto-fill": "auto-fill"
}, B = {
  ...e,
  ...a,
  "min-content": "min-content",
  "max-content": "max-content",
  fr: "1fr"
}, $ = {
  auto: "auto",
  1: "1",
  2: "2",
  3: "3",
  4: "4",
  5: "5",
  6: "6",
  7: "7",
  8: "8",
  9: "9",
  10: "10",
  11: "11",
  12: "12",
  13: "13",
  14: "14",
  15: "15",
  16: "16",
  17: "17"
}, o = {
  auto: "auto",
  1: "1",
  2: "2",
  3: "3",
  4: "4",
  5: "5",
  6: "6",
  7: "7",
  8: "8",
  9: "9",
  10: "10",
  11: "11",
  12: "12",
  13: "13",
  14: "14",
  15: "15",
  16: "16",
  full: "full"
}, h = {
  "0.75": `${0.8 * 0.75}rem`,
  1: `${0.8 * 1}rem`,
  "1.25": `${0.8 * 1.25}rem`,
  "1.5": `${0.8 * 1.5}rem`,
  "1.75": `${0.8 * 1.75}rem`,
  2: `${0.8 * 2}rem`,
  "2.25": `${0.8 * 2.25}rem`,
  "2.5": `${0.8 * 2.5}rem`,
  "2.75": `${0.8 * 2.75}rem`,
  3: `${0.8 * 3}rem`,
  "3.25": `${0.8 * 3.25}rem`,
  "3.5": `${0.8 * 3.5}rem`,
  "3.75": `${0.8 * 3.75}rem`,
  4: `${0.8 * 4}rem`,
  "4.5": `${0.8 * 4.5}rem`,
  5: `${0.8 * 5}rem`,
  "5.5": `${0.8 * 5.5}rem`,
  6: `${0.8 * 6}rem`,
  "6.5": `${0.8 * 6.5}rem`,
  7: `${0.8 * 7}rem`,
  "7.5": `${0.8 * 7.5}rem`,
  8: `${0.8 * 8}rem`,
  9: `${0.8 * 9}rem`,
  10: `${0.8 * 10}rem`,
  11: `${0.8 * 11}rem`,
  12: `${0.8 * 12}rem`,
  13: `${0.8 * 13}rem`,
  14: `${0.8 * 14}rem`,
  15: `${0.8 * 15}rem`,
  16: `${0.8 * 16}rem`,
  18: `${0.8 * 18}rem`,
  20: `${0.8 * 20}rem`,
  22: `${0.8 * 22}rem`,
  24: `${0.8 * 24}rem`,
  26: `${0.8 * 26}rem`,
  28: `${0.8 * 28}rem`,
  30: `${0.8 * 30}rem`,
  32: `${0.8 * 32}rem`,
  inherit: "inherit"
}, T = {
  "0%": "0em",
  "2%": "0.02em",
  "4%": "0.04em",
  "6%": "0.06em",
  "8%": "0.08em",
  "10%": "0.1em",
  "12%": "0.12em",
  "14%": "0.14em",
  "16%": "0.16em",
  "18%": "0.18em",
  "20%": "0.2em"
}, b = {
  "90%": "0.9",
  "100%": "1",
  "110%": "1.1",
  "120%": "1.2",
  "130%": "1.3",
  "140%": "1.4",
  "150%": "1.5",
  "160%": "1.6",
  "170%": "1.7",
  "180%": "1.8",
  "190%": "1.9",
  "200%": "2"
}, x = {
  none: "none",
  1: "1",
  2: "2",
  3: "3",
  4: "4",
  5: "5",
  6: "6",
  7: "7",
  8: "8"
}, k = {
  t: "to top",
  tr: "to top right",
  r: "to right",
  br: "to bottom right",
  b: "to bottom",
  bl: "to bottom left",
  l: "to left",
  tl: "to top left"
};
export {
  f as aspectRatioTheme,
  d as borderRadiusTheme,
  i as borderTypeTheme,
  s as borderWidthTheme,
  u as degreeTheme,
  g as durationTheme,
  p as flexItemTheme,
  h as fontSizeTheme,
  r as gapTheme,
  $ as gridLineTheme,
  o as gridSpanTheme,
  S as gridTemplateTrackTheme,
  B as gridTrackMinMaxTheme,
  T as letterSpacingTheme,
  x as lineClampTheme,
  b as lineHeightTheme,
  k as linearGradientTheme,
  c as opacityTheme,
  a as percentageTheme,
  m as positioningTheme,
  l as scaleTheme,
  t as sizingTheme,
  e as spacingTheme,
  n as zIndexTheme
};
