// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Atoms/Button.js
import { jsx as y } from "react/jsx-runtime";
import $ from "../../utils/classNamesObjectToString.js";
import j from "../../utils/deepMerge.js";
const S = ({
  className: e,
  style: o,
  title: r,
  id: s,
  type: a = "button",
  children: n,
  theme: t = "filled",
  onClick: l,
  disabled: c,
  noDefaultStyle: i,
  hover: d,
  htmlAttributes: p,
  ...m
}) => {
  const u = m, f = i ? {} : {
    size: "sz-auto",
    color: "c-transparent",
    padding: "p-2",
    margin: "m-0",
    justifyContent: "jc-center",
    alignItems: "ai-center",
    alignContent: "ac-center",
    wrap: "flex-nowrap",
    direction: "flex-row",
    gap: "g-1",
    borderRadius: "rad-3",
    cursor: "cs-pointer",
    disabledStyle: {
      opacity: "disabled:oc-0.5",
      cursor: "disabled:cs-default"
    }
  }, { ...b } = p || {}, g = d ? "hover" : "", x = t ? `${t !== "filled" ? t : ""}` : "";
  return /* @__PURE__ */ y(
    "button",
    {
      type: a,
      className: `flex ${x} ${g} ${$(
        j(f, u)
      )} ${e || ""}`,
      onClick: l,
      disabled: c,
      style: o,
      id: s,
      title: r,
      ...b,
      children: n
    }
  );
};
export {
  S as default
};
