// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Atoms/ButtonWithRef.js
"use client";
import { jsx as $ } from "react/jsx-runtime";
import j from "../../utils/classNamesObjectToString.js";
import { forwardRef as C } from "react";
import N from "../../utils/deepMerge.js";
const v = C(
  ({
    type: r = "button",
    className: o,
    style: a,
    title: s,
    id: n,
    children: i,
    theme: t = "filled",
    onClick: e,
    disabled: l,
    hover: c,
    noDefaultStyle: d,
    htmlAttributes: f,
    ...p
  }, m) => {
    const u = d ? {} : {
      size: "sz-auto",
      color: "c-transparent",
      padding: "p-2",
      margin: "m-0",
      justifyContent: "jc-center",
      alignItems: "ai-center",
      alignContent: "ac-center",
      wrap: "flex-nowrap",
      direction: "flex-row",
      gap: "g-1",
      borderRadius: "rad-3",
      cursor: "cs-pointer",
      disabledStyle: { cursor: "disabled:cs-default" }
    }, { ...g } = f || {}, b = c ? "hover" : "", w = t ? `${t !== "filled" ? t : ""}` : "";
    return /* @__PURE__ */ $(
      "button",
      {
        type: r,
        className: `flex ${w} ${b} ${j(
          N(u, p)
        )} ${o || ""}`,
        onClick: (x) => {
          x.stopPropagation(), e && e();
        },
        disabled: l,
        style: a,
        id: n,
        title: s,
        ...g,
        ref: m,
        children: i
      }
    );
  }
);
export {
  v as default
};
