// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Atoms/InputWithRef.js
"use client";
import { jsx as z } from "react/jsx-runtime";
import N from "../../utils/classNamesObjectToString.js";
import P from "../../utils/deepMerge.js";
import { forwardRef as S } from "react";
const H = S(
  ({
    type: t,
    className: r,
    style: o,
    title: e,
    id: s,
    name: a,
    value: i,
    defaultValue: n,
    checked: m,
    disabled: p,
    placeholder: f,
    required: c,
    onChange: l,
    onBlur: u,
    onKeyDown: d,
    noDefaultStyle: b,
    htmlAttributes: g,
    ...y
  }, j) => {
    const x = y, R = b ? {} : {
      size: "sz-auto",
      color: "c-neutral-obj",
      padding: "p-0",
      margin: "m-0",
      borderRadius: "rad-0",
      font: "f-2",
      fontFamily: "ff-primary",
      cursor: "cs-text"
    }, { ...w } = g || {};
    return /* @__PURE__ */ z(
      "input",
      {
        className: `input plain ${N(
          P(R, x)
        )} ${r || ""}`,
        type: t,
        id: s,
        name: a,
        value: i,
        onChange: l,
        onBlur: u,
        onKeyDown: d,
        checked: m,
        disabled: p,
        placeholder: f,
        defaultValue: n,
        required: c,
        style: o,
        title: e,
        ref: j,
        ...w
      }
    );
  }
);
export {
  H as default
};
