// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Atoms/Input.js
import { jsx as S } from "react/jsx-runtime";
import $ from "../../utils/classNamesObjectToString.js";
import w from "../../utils/deepMerge.js";
const I = ({
  type: t,
  id: r,
  name: o,
  className: s,
  style: e,
  title: a,
  value: n,
  defaultValue: i,
  checked: p,
  disabled: m,
  placeholder: c,
  required: l,
  step: u,
  maxLength: f,
  onChange: d,
  onBlur: b,
  onKeyDown: g,
  noDefaultStyle: y,
  htmlAttributes: j,
  ...x
}) => {
  const z = x, N = y ? {} : {
    size: "sz-auto",
    color: "c-neutral-obj",
    padding: "p-0",
    margin: "m-0",
    borderRadius: "rad-0",
    font: "f-2",
    fontFamily: "ff-primary",
    cursor: "cs-text"
  }, { ...P } = j || {};
  return /* @__PURE__ */ S(
    "input",
    {
      className: `input plain ${$(
        w(N, z)
      )} ${s || ""}`,
      type: t,
      id: r,
      name: o,
      value: n,
      step: u,
      onChange: d,
      onBlur: b,
      onKeyDown: g,
      checked: p,
      disabled: m,
      placeholder: c,
      defaultValue: i,
      required: l,
      style: e,
      title: a,
      size: f,
      ...P
    }
  );
};
export {
  I as default
};
