// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Atoms/ScrollWithRef.js
"use client";
import { jsx as u } from "react/jsx-runtime";
import { forwardRef as g } from "react";
import h from "../../utils/classNamesObjectToString.js";
import b from "../../utils/deepMerge.js";
const S = g(
  ({
    children: e,
    className: t,
    style: o,
    title: l,
    id: s,
    isWheelDownToRight: a,
    noDefaultStyle: i,
    htmlAttributes: c,
    ...f
  }, m) => {
    const n = i ? {} : {
      scrollBar: "scr-visible",
      size: "sz-full",
      minHeight: "mh-0",
      minWidth: "mw-0",
      overflow: "ovf-auto",
      color: "c-transparent",
      padding: "p-0",
      margin: "m-0",
      borderRadius: "rad-0",
      direction: "flex-col"
    }, { ...d } = c || {}, p = (r) => {
      a && r.currentTarget && (r.preventDefault(), r.currentTarget.scrollLeft += r.deltaY);
    };
    return /* @__PURE__ */ u(
      "div",
      {
        className: `scroll flex ${h(
          b(n, f)
        )} ${t || ""}`,
        onWheel: p,
        style: o,
        id: s,
        title: l,
        ...d,
        ref: m,
        children: e
      }
    );
  }
);
export {
  S as default
};
