// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Atoms/Textarea.js
import { jsx as N } from "react/jsx-runtime";
import P from "../../utils/classNamesObjectToString.js";
import S from "../../utils/deepMerge.js";
const A = ({
  id: t,
  name: r,
  className: e,
  rows: o,
  cols: s,
  maxLength: a,
  style: i,
  value: n,
  defaultValue: m,
  disabled: p,
  placeholder: c,
  required: l,
  onChange: f,
  onBlur: u,
  onKeyDown: d,
  noDefaultStyle: x,
  htmlAttributes: b,
  ...g
}) => {
  const y = g, z = x ? {} : {
    size: "sz-auto",
    color: "c-neutral-obj",
    padding: "p-0",
    margin: "m-0",
    borderRadius: "rad-0",
    font: "f-1.75",
    fontFamily: "ff-primary",
    cursor: "cs-text",
    resize: "resize"
  }, { ...j } = b || {};
  return /* @__PURE__ */ N(
    "textarea",
    {
      className: `input plain ${P(
        S(z, y)
      )} ${e || ""}`,
      id: t,
      name: r,
      value: n,
      onChange: f,
      onBlur: u,
      onKeyDown: d,
      disabled: p,
      placeholder: c,
      defaultValue: m,
      required: l,
      style: i,
      rows: o,
      cols: s,
      maxLength: a,
      ...j
    }
  );
};
export {
  A as default
};
