// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Atoms/TextareaWithRef.js
import { jsx as w } from "react/jsx-runtime";
import { forwardRef as N } from "react";
import P from "../../utils/classNamesObjectToString.js";
import S from "../../utils/deepMerge.js";
const F = N(
  ({
    id: r,
    name: t,
    className: e,
    rows: o,
    cols: s,
    maxLength: a,
    style: i,
    value: m,
    defaultValue: f,
    disabled: n,
    placeholder: p,
    required: c,
    onChange: l,
    onBlur: d,
    onKeyDown: u,
    noDefaultStyle: x,
    htmlAttributes: b,
    ...g
  }, y) => {
    const z = g, j = x ? {} : {
      size: "sz-auto",
      color: "c-neutral-obj",
      padding: "p-0",
      margin: "m-0",
      borderRadius: "rad-0",
      font: "f-1.75",
      fontFamily: "ff-primary",
      cursor: "cs-text",
      resize: "resize"
    }, { ...R } = b || {};
    return /* @__PURE__ */ w(
      "textarea",
      {
        className: `input plain ${P(
          S(j, z)
        )} ${e || ""}`,
        ref: y,
        id: r,
        name: t,
        value: m,
        onChange: l,
        onBlur: d,
        onKeyDown: u,
        disabled: n,
        placeholder: p,
        defaultValue: f,
        required: c,
        style: i,
        rows: o,
        cols: s,
        maxLength: a,
        ...R
      }
    );
  }
);
export {
  F as default
};
