// @flowwwkr/design-system-tailwind@0.12.12/dist/tailwind/plugins/flexUtilites.js
import { sizingTheme as a, flexItemTheme as s, gapTheme as l } from "../themes/index.js";
import i from "tailwindcss/plugin";
const f = i(function({
  addUtilities: n,
  matchUtilities: t
}) {
  n({
    ".jc-start": {
      "justify-content": "flex-start"
    },
    ".jc-end": {
      "justify-content": "flex-end"
    },
    ".jc-center": {
      "justify-content": "center"
    },
    ".jc-between": {
      "justify-content": "space-between"
    },
    ".jc-around": {
      "justify-content": "space-around"
    },
    ".jc-evenly": {
      "justify-content": "space-evenly"
    },
    ".ai-start": {
      "align-items": "flex-start"
    },
    ".ai-end": {
      "align-items": "flex-end"
    },
    ".ai-center": {
      "align-items": "center"
    },
    ".ai-baseline": {
      "align-items": "baseline"
    },
    ".ai-stretch": {
      "align-items": "stretch"
    },
    ".ac-start": {
      "align-content": "flex-start"
    },
    ".ac-end": {
      "align-content": "flex-end"
    },
    ".ac-center": {
      "align-content": "center"
    },
    ".ac-between": {
      "align-content": "space-between"
    },
    ".ac-around": {
      "align-content": "space-around"
    },
    ".ac-stretch": {
      "align-content": "stretch"
    },
    ".as-start": {
      "align-self": "flex-start"
    },
    ".as-end": {
      "align-self": "flex-end"
    },
    ".as-center": {
      "align-self": "center"
    },
    ".as-baseline": {
      "align-self": "baseline"
    },
    ".as-stretch": {
      "align-self": "stretch"
    },
    ".as-auto": {
      "align-self": "auto"
    }
  }), t(
    {
      basis: (e) => ({
        "flex-basis": e
      })
    },
    {
      values: a
    }
  ), t(
    {
      grow: (e) => ({
        "flex-grow": e
      }),
      shrink: (e) => ({
        "flex-shrink": e
      })
    },
    {
      values: s
    }
  ), t(
    {
      g: (e) => ({
        gap: e
      })
    },
    {
      values: l
    }
  );
});
export {
  f as flexUtilities
};
