// @flowwwkr/design-system-tailwind@0.12.12/dist/tailwind/plugins/sizingUtilities.js
import n from "tailwindcss/plugin";
import { sizingTheme as e } from "../themes/index.js";
const t = n(function({ matchUtilities: i }) {
  i(
    {
      sz: (h) => ({
        width: h === "screen" ? "100vw" : h,
        height: h === "screen" ? "100vh" : h
      }),
      // w: (value) => ({
      //   width: value === 'screen' ? '100vw' : value,
      // }),
      // h: (value) => ({
      //   height: value === 'screen' ? '100vh' : value,
      // }),
      xw: (h) => ({
        "max-width": h === "screen" ? "100vw" : h
      }),
      xh: (h) => ({
        "max-height": h === "screen" ? "100vh" : h
      }),
      mw: (h) => ({
        "min-width": h === "screen" ? "100vw" : h
      }),
      mh: (h) => ({
        "min-height": h === "screen" ? "100vh" : h
      })
    },
    {
      values: e
    }
  ), i(
    {
      w: (h) => ({
        width: h === "screen" ? "100vw" : h
      }),
      h: (h) => ({
        height: h === "screen" ? "100vh" : h
      })
    },
    {
      values: e
    }
  );
});
export {
  t as sizingUtilities
};
