// @flowwwkr/design-system-tailwind@0.12.12/dist/tailwind/plugins/typographyUtilities.js
import { fontSizeTheme as i, letterSpacingTheme as n, lineHeightTheme as r, lineClampTheme as f } from "../themes/index.js";
import l from "tailwindcss/plugin";
const m = l(function({
  addUtilities: t,
  matchUtilities: o
}) {
  t({
    ".ff-pretendard": {
      "font-family": 'var(--font-pretendard), "system-ui", "sans-serif", "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"'
    },
    ".ff-suit": {
      "font-family": 'var(--font-suit), "system-ui", "sans-serif", "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"'
    },
    ".ff-noto-sans-kr": {
      "font-family": 'var(--font-noto-sans-kr), "system-ui", "sans-serif", "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"'
    },
    ".ff-freesentation": {
      "font-family": 'var(--font-freesentation), "system-ui", "sans-serif", "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"'
    },
    ".ff-primary": {
      "font-family": 'var(--font-primary), "system-ui", "sans-serif", "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"'
    },
    ".ff-secondary": {
      "font-family": 'var(--font-secondary), "system-ui", "sans-serif", "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"'
    },
    ".ff-tertiary": {
      "font-family": 'var(--font-tertiary), "system-ui", "sans-serif", "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"'
    },
    ".ff-inherit": {
      "font-family": "inherit"
    }
  }), o(
    {
      f: (e) => parseFloat(e) >= 2.4 ? {
        "font-size": e,
        "font-weight": "700",
        "line-height": "normal"
      } : e === "inherit" ? {
        "font-size": e,
        "font-weight": "inherit",
        "line-height": "inherit"
      } : {
        "font-size": e,
        "font-weight": "400",
        "line-height": "normal"
      }
    },
    { values: i }
  ), o(
    {
      fs: (e) => e === "inherit" ? {
        "font-size": e
      } : {
        "font-size": e
      }
    },
    { values: i }
  ), t({
    ".fw-100": {
      "font-weight": "100"
    },
    ".fw-200": {
      "font-weight": "200"
    },
    ".fw-300": {
      "font-weight": "300"
    },
    ".fw-400": {
      "font-weight": "400"
    },
    ".fw-500": {
      "font-weight": "500"
    },
    ".fw-600": {
      "font-weight": "600"
    },
    ".fw-700": {
      "font-weight": "700"
    },
    ".fw-800": {
      "font-weight": "800"
    },
    ".fw-900": {
      "font-weight": "900"
    },
    ".fw-inherit": {
      "font-weight": "inherit"
    }
  }), o(
    {
      ls: (e) => ({
        "letter-spacing": e
      })
    },
    {
      values: n,
      supportsNegativeValues: !0
    }
  ), o(
    {
      lh: (e) => ({
        "line-height": e
      })
    },
    { values: r }
  ), o(
    {
      lc: (e) => e === "none" ? {
        overflow: "visible",
        display: "block",
        "-webkit-box-orient": "horizontal",
        "-webkit-line-clamp": "none"
      } : {
        overflow: "hidden",
        display: "-webkit-box",
        "-webkit-box-orient": "vertical",
        "-webkit-line-clamp": e
      }
    },
    {
      values: f
    }
  ), t({
    ".ta-left": {
      "text-align": "left"
    },
    ".ta-center": {
      "text-align": "center"
    },
    ".ta-right": {
      "text-align": "right"
    },
    ".ta-justify": {
      "text-align": "justify"
    },
    ".ta-start": {
      "text-align": "start"
    },
    ".ta-end": {
      "text-align": "end"
    }
  }), t({
    ".va-baseline": {
      "vertical-align": "baseline"
    },
    ".va-top": {
      "vertical-align": "top"
    },
    ".va-middle": {
      "vertical-align": "middle"
    },
    ".va-bottom": {
      "vertical-align": "bottom"
    },
    ".va-text-top": {
      "vertical-align": "text-top"
    },
    ".va-text-bottom": {
      "vertical-align": "text-bottom"
    },
    ".va-sub": {
      "vertical-align": "sub"
    },
    ".va-super": {
      "vertical-align": "super"
    }
  });
});
export {
  m as typographyUtilities
};
