// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Molecules/ArrayInput.js
import { jsx as V, jsxs as v } from "react/jsx-runtime";
import g from "./TextInput.js";
import H from "../Atoms/Flex.js";
import J from "../Atoms/Button.js";
import d from "./ThemeNumberInput.js";
import a from "../Icons/IconCircleMinus.js";
import C from "../Icons/IconCirclePlus.js";
import z from "../../utils/deepMerge.js";
import { useState as n, useEffect as tt } from "react";
const wt = ({
  id: k,
  name: o,
  required: M,
  placeholder: T,
  values: c,
  defaultValues: K,
  maxItems: y,
  minItems: s = 1,
  // 기본 최소 개수 설정
  showAddButton: L = !0,
  showRemoveButton: O = !0,
  addIcon: _,
  removeIcon: B,
  type: w = "text",
  align: Q = "vertical",
  size: D = "md",
  color: E = "neutral",
  state: F = "default",
  theme: I = "outlined",
  textInputProps: U,
  numberInputProps: X,
  rootStyle: Y,
  itemWrapperStyle: Z,
  addIconStyle: $,
  removeIconStyle: b,
  disabled: p,
  // validate,
  onAdd: e,
  onRemove: S,
  onChange: l
}) => {
  var W, G;
  const [u, f] = n(
    c || K || []
  ), q = {
    sm: {
      controlIconStyle: { size: "sz-1.75" }
    },
    md: {
      controlIconStyle: { size: "sz-2.25" }
    },
    lg: {
      controlIconStyle: { size: "sz-2.5" }
    }
  };
  tt(() => {
    if (c) {
      if (c.length < s) {
        const t = [...c];
        for (; t.length < s; )
          t.push(w === "text" ? "" : 0);
        f(t), l == null || l(t);
      }
    } else if (u.length < s) {
      const t = [...u];
      for (; t.length < s; )
        t.push(w === "text" ? "" : 0);
      f(t), l == null || l(t);
    }
  }, [s, c, u, w, l]);
  const m = H, x = {
    position: "relative",
    width: "w-auto",
    height: "h-auto",
    direction: Q === "vertical" ? "flex-col" : "flex-row",
    gap: "g-1"
  }, R = {
    gap: "g-0.5"
  }, N = {
    size: (G = (W = q[D]) == null ? void 0 : W.controlIconStyle) == null ? void 0 : G.size
  }, A = (t) => {
    if (!p)
      if (c) {
        if (y && c.length >= y) return;
        const r = [...c];
        r.splice(t + 1, 0, w === "text" ? "" : 0), f(r), e == null || e(r[t + 1], t + 1), l == null || l(r);
      } else {
        if (y && u.length >= y) return;
        const r = [...u];
        r.splice(t + 1, 0, w === "text" ? "" : 0), f(r), e == null || e(r[t + 1], t + 1), l == null || l(r);
      }
  }, h = (t) => {
    if (c) {
      if (p || c.length <= s) return;
      const r = c.filter((i, j) => j !== t);
      f(r), S == null || S(c[t], t), l == null || l(r);
    } else {
      if (p || u.length <= s) return;
      const r = u.filter((i, j) => j !== t);
      f(r), S == null || S(u[t], t), l == null || l(r);
    }
  }, P = (t, r) => {
    if (c) {
      const i = [...c];
      i[t] = r, f(i), l == null || l(i);
    } else {
      const i = [...u];
      i[t] = r, f(i), l == null || l(i);
    }
  };
  return /* @__PURE__ */ V(m, { ...z(x, Y || {}), noDefaultStyle: !0, children: (c || u).map((t, r) => /* @__PURE__ */ v(
    H,
    {
      ...z(R, Z || {}),
      children: [
        w === "text" ? /* @__PURE__ */ V(
          g,
          {
            id: k,
            name: o,
            required: M,
            placeholder: T,
            size: D,
            color: E,
            state: F,
            theme: I,
            disabled: p,
            value: t,
            onChange: (i) => P(r, i),
            ...U
          }
        ) : /* @__PURE__ */ V(
          d,
          {
            id: k,
            name: o,
            required: M,
            placeholder: T,
            size: D,
            color: E,
            state: F,
            theme: I,
            disabled: p,
            value: t,
            onChange: (i) => P(r, i),
            ...X
          }
        ),
        /* @__PURE__ */ V(
          J,
          {
            visibility: L ? "visible" : "hidden",
            padding: "p-0",
            onClick: () => A(r),
            disabled: p,
            noDefaultStyle: !0,
            children: _ || /* @__PURE__ */ V(
              C,
              {
                ...z(
                  { ...N },
                  $ || {}
                )
              }
            )
          }
        ),
        /* @__PURE__ */ V(
          J,
          {
            visibility: O ? "visible" : "hidden",
            padding: "p-0",
            onClick: () => h(r),
            disabled: p,
            noDefaultStyle: !0,
            children: B || /* @__PURE__ */ V(
              a,
              {
                ...z(
                  { ...N },
                  b || {}
                )
              }
            )
          }
        )
      ]
    },
    r
  )) });
};
export {
  wt as default
};
