// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Molecules/CheckBox.js
import { jsxs as I, jsx as S } from "react/jsx-runtime";
import { useState as x, useEffect as R } from "react";
import P from "../Icons/IconCheck.js";
import V from "../Icons/IconIndeterminate.tsx.js";
import u from "../../utils/deepMerge.js";
import A from "../Atoms/Label.js";
const N = ({
  disabled: t = !1,
  checked: o,
  indeterminate: p = !1,
  onChange: n,
  color: e = "neutral",
  size: y = "md",
  theme: i = "filled",
  rootStyle: m,
  indicatorStyle: C
}) => {
  const [l, s] = x(o ?? !1);
  R(() => {
    o !== void 0 && s(o);
  }, [o]);
  const h = () => o !== void 0 ? o ? "on" : "off" : l ? "on" : "off", j = (r) => {
    if (r.stopPropagation(), t) return;
    const b = o !== void 0 ? !o : !l;
    o === void 0 && s(b), n && n(b);
  }, d = {
    sm: {
      rootStyle: {
        width: "w-2",
        height: "h-2",
        borderRadius: "rad-0.25"
      },
      indicatorStyle: { width: "w-full", height: "h-full" }
    },
    md: {
      rootStyle: {
        width: "w-3",
        height: "h-3",
        borderRadius: "rad-0.5"
      },
      indicatorStyle: { width: "w-full", height: "h-full" }
    },
    lg: {
      rootStyle: {
        width: "w-4.5",
        height: "h-4.5",
        borderRadius: "rad-1"
      },
      indicatorStyle: { width: "w-full", height: "h-full" }
    }
  }, c = {
    neutral: {
      filled: {
        on: {
          rootStyle: { color: "c-neutral-obj", borderColor: "bc-neutral-obj" },
          indicatorStyle: { color: "c-neutral-sf" }
        },
        off: {
          rootStyle: { color: "c-neutral-sf", borderColor: "bc-neutral-sf-04" },
          indicatorStyle: { color: "c-neutral-obj" }
        }
      },
      outlined: {
        on: {
          rootStyle: { color: "c-neutral-sf", borderColor: "bc-neutral-obj" },
          indicatorStyle: { color: "c-neutral-obj" }
        },
        off: {
          rootStyle: { color: "c-neutral-sf", borderColor: "bc-neutral-sf-04" },
          indicatorStyle: { color: "c-neutral-sf" }
        }
      }
    },
    primary: {
      filled: {
        on: {
          rootStyle: { color: "c-primary-obj", borderColor: "bc-primary-obj" },
          indicatorStyle: { color: "c-primary-sf" }
        },
        off: {
          rootStyle: { color: "c-primary-sf", borderColor: "bc-primary-sf-04" },
          indicatorStyle: { color: "c-primary-obj" }
        }
      },
      outlined: {
        on: {
          rootStyle: { color: "c-primary-sf", borderColor: "bc-primary-obj" },
          indicatorStyle: { color: "c-primary-obj" }
        },
        off: {
          rootStyle: { color: "c-primary-sf", borderColor: "bc-primary-sf-04" },
          indicatorStyle: { color: "c-primary-sf" }
        }
      }
    },
    secondary: {
      filled: {
        on: {
          rootStyle: {
            color: "c-secondary-obj",
            borderColor: "bc-secondary-obj"
          },
          indicatorStyle: { color: "c-secondary-sf" }
        },
        off: {
          rootStyle: {
            color: "c-secondary-sf",
            borderColor: "bc-secondary-sf-04"
          },
          indicatorStyle: { color: "c-secondary-obj" }
        }
      },
      outlined: {
        on: {
          rootStyle: {
            color: "c-secondary-sf",
            borderColor: "bc-secondary-obj"
          },
          indicatorStyle: { color: "c-secondary-obj" }
        },
        off: {
          rootStyle: {
            color: "c-secondary-sf",
            borderColor: "bc-secondary-sf-04"
          },
          indicatorStyle: { color: "c-secondary-sf" }
        }
      }
    },
    tertiary: {
      filled: {
        on: {
          rootStyle: {
            color: "c-tertiary-obj",
            borderColor: "bc-tertiary-obj"
          },
          indicatorStyle: { color: "c-tertiary-sf" }
        },
        off: {
          rootStyle: {
            color: "c-tertiary-sf",
            borderColor: "bc-tertiary-sf-04"
          },
          indicatorStyle: { color: "c-tertiary-obj" }
        }
      },
      outlined: {
        on: {
          rootStyle: {
            color: "c-tertiary-sf",
            borderColor: "bc-tertiary-obj"
          },
          indicatorStyle: { color: "c-tertiary-obj" }
        },
        off: {
          rootStyle: {
            color: "c-tertiary-sf",
            borderColor: "bc-tertiary-sf-04"
          },
          indicatorStyle: { color: "c-tertiary-sf" }
        }
      }
    }
    // Additional color themes omitted for brevity
  }, a = h(), w = {
    ...d[y].rootStyle,
    cursor: "cs-pointer",
    display: "block",
    opacity: t ? "oc-0.5" : "oc-1",
    color: c[e][i][a].rootStyle.color,
    borderWidth: "bw-0.125",
    borderStyle: "bs-solid",
    borderColor: c[e][i][a].rootStyle.borderColor,
    position: "relative"
  }, g = {
    ...d[y].indicatorStyle,
    color: c[e][i][a].indicatorStyle.color,
    opacity: t ? "oc-0.5" : "oc-1",
    position: "absolute"
  }, v = A, f = p ? V : (o !== void 0 ? o : l) ? P : null;
  return /* @__PURE__ */ I(
    v,
    {
      ...u(w, m || {}),
      noDefaultStyle: !0,
      htmlAttributes: {
        onClick: (r) => r.stopPropagation()
        // 여기서도 부모 클릭 이벤트 전파 방지
      },
      children: [
        f ? /* @__PURE__ */ S(
          f,
          {
            ...u(g, C || {}),
            noDefaultStyle: !0,
            htmlAttributes: {
              onClick: (r) => r.stopPropagation()
              // 여기서도 부모 클릭 이벤트 전파 방지
            }
          }
        ) : null,
        /* @__PURE__ */ S(
          "input",
          {
            type: "checkbox",
            checked: o !== void 0 ? o : l,
            disabled: t,
            onChange: j,
            className: "hidden",
            onClick: (r) => r.stopPropagation()
          }
        )
      ]
    }
  );
};
export {
  N as default
};
