// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Molecules/MultiSelect.js
"use client";
import { jsxs as w, jsx as u } from "react/jsx-runtime";
import { useState as m, useRef as fr, useEffect as mr } from "react";
import st from "../Atoms/WrapperWithRef.js";
import yt from "../Atoms/Button.js";
import br from "../Atoms/Text.js";
import dt from "../Atoms/ScrollWithRef.js";
import hr from "../Icons/IconArrow.js";
import at from "../Atoms/Flex.js";
import y from "../../utils/deepMerge.js";
import gt from "../Icons/IconCheck.js";
import Ir from "../Atoms/Wrapper.js";
import { findScrollableOrHiddenParent as ut } from "../../utils/client/index.js";
const Dt = ({
  id: xr,
  name: wr,
  required: jr,
  limit: D,
  options: b,
  placeholder: Tr,
  value: n,
  defaultValue: H,
  size: t = "md",
  color: a = "neutral",
  state: c = "default",
  theme: i = "outlined",
  disabled: j,
  onChange: d,
  rootStyle: vr,
  triggerStyle: Wr,
  triggerTextStyle: Dr,
  triggerIconStyle: Hr,
  menuWrapperStyle: Nr,
  menuStyle: kr,
  optionStyle: Cr,
  optionTextStyle: zr,
  optionIconStyle: Or,
  scrollNudgerStyle: N,
  scrollIconStyle: k
}) => {
  var J, $, K, q, G, Q, Y, Z, _, ee, re, te, oe, le, ce, ne, ie, se, ye, de, ae, ge, ue, Se, pe, fe, me, be, he, Ie, xe, we, je, Te, ve, We, De, He, Ne, ke, Ce, ze, Oe, Re, Ve, Me, Xe, Be, Ee, Ae, Le, Ue, Pe, Fe, Je, $e, Ke, qe, Ge, Qe, Ye, Ze, _e, er, rr, tr, or, lr, cr, nr, ir, sr, yr, dr, ar, gr, ur, Sr, pr;
  const [I, C] = m(() => n !== void 0 ? n : H !== void 0 ? H : []), [x, T] = m(!1), [Rr, Vr] = m(!1), [Mr, Xr] = m(!1), [z, O] = m(
    null
  ), [R, V] = m(!1), [M, X] = m(-1), S = fr(null), h = fr(null), o = {
    sm: {
      rootStyle: { width: "w-24" },
      triggerStyle: { height: "h-2.75", paddingX: "px-1", gap: "g-1" },
      triggerTextStyle: { font: "f-1.5" },
      triggerIconStyle: {
        size: "sz-1.75",
        minWidth: "mw-1.75",
        minHeight: "mh-1.75"
      },
      menuWrapperStyle: { padding: "p-0.5" },
      menuStyle: { maxHeight: "xh-22", gap: "g-0.5" },
      optionStyle: { height: "h-2.75", paddingX: "px-1", gap: "g-1" },
      optionTextStyle: { font: "f-1.5" },
      optionIconStyle: { size: "sz-1.75" },
      scrollIconStyle: { size: "sz-1.75" }
    },
    md: {
      rootStyle: { width: "w-32" },
      triggerStyle: { height: "h-4.25", paddingX: "px-1.5", gap: "g-1.5" },
      triggerTextStyle: { font: "f-1.75" },
      triggerIconStyle: {
        size: "sz-2.25",
        minWidth: "mw-2.25",
        minHeight: "mh-2.25"
      },
      menuWrapperStyle: { padding: "p-0.5" },
      menuStyle: { maxHeight: "xh-30", gap: "g-0.5" },
      optionStyle: { height: "h-4.25", paddingX: "px-1.5", gap: "g-1.5" },
      optionTextStyle: { font: "f-1.75" },
      optionIconStyle: { size: "sz-2.25" },
      scrollIconStyle: { size: "sz-2.25" }
    },
    lg: {
      rootStyle: { width: "w-40" },
      triggerStyle: { height: "h-4.5", paddingX: "px-1.5", gap: "g-1.5" },
      triggerTextStyle: { font: "f-2" },
      triggerIconStyle: {
        size: "sz-2.5",
        minWidth: "mw-2.5",
        minHeight: "mh-2.5"
      },
      menuWrapperStyle: { padding: "p-0.5" },
      menuStyle: { maxHeight: "xh-36", gap: "g-0.5" },
      optionStyle: { height: "h-4.5", paddingX: "px-1.5", gap: "g-1.5" },
      optionTextStyle: { font: "f-2" },
      optionIconStyle: { size: "sz-2.5" },
      scrollIconStyle: { size: "sz-2.5" }
    }
  }, p = {
    error: {
      filled: {
        triggerStyle: { color: "c-error-sf-01" },
        triggerTextStyle: { color: "c-error-obj-01" },
        triggerIconStyle: { color: "c-error-sf-04" }
      },
      outlined: {
        triggerStyle: { borderColor: "bc-error-mid" },
        triggerTextStyle: { color: "c-error-mid" },
        triggerIconStyle: { color: "c-error-obj-01" }
      }
    },
    warning: {
      filled: {
        triggerStyle: { color: "c-warning-sf-01" },
        triggerTextStyle: { color: "c-warning-obj-01" },
        triggerIconStyle: { color: "c-warning-sf-04" }
      },
      outlined: {
        triggerStyle: { borderColor: "bc-warning-mid" },
        triggerTextStyle: { color: "c-warning-mid" },
        triggerIconStyle: { color: "c-warning-obj-01" }
      }
    },
    success: {
      filled: {
        triggerStyle: { color: "c-success-sf-01" },
        triggerTextStyle: { color: "c-success-obj-01" },
        triggerIconStyle: { color: "c-success-sf-04" }
      },
      outlined: {
        triggerStyle: { borderColor: "bc-success-mid" },
        triggerTextStyle: { color: "c-success-mid" },
        triggerIconStyle: { color: "c-success-obj-01" }
      }
    },
    perfect: {
      filled: {
        triggerStyle: { color: "c-perfect-sf-01" },
        triggerTextStyle: { color: "c-perfect-obj-01" },
        triggerIconStyle: { color: "c-perfect-sf-04" }
      },
      outlined: {
        triggerStyle: { borderColor: "bc-perfect-mid" },
        triggerTextStyle: { color: "c-perfect-mid" },
        triggerIconStyle: { color: "c-perfect-obj-01" }
      }
    }
  }, g = {
    primary: {
      filled: {
        ...c === "default" ? {
          triggerStyle: { color: "c-primary-sf-01" },
          triggerTextStyle: { color: "c-primary-obj" },
          triggerIconStyle: { color: "c-primary-obj-02" }
        } : (J = p[c]) == null ? void 0 : J.filled,
        menuWrapperStyle: {
          borderColor: "bc-primary-mid",
          color: "c-primary-sf"
        },
        menuStyle: { color: "c-primary-sf" },
        optionStyle: { color: "c-primary-sf" },
        optionTextStyle: { color: "c-primary-obj" },
        optionIconStyle: { color: "c-primary-obj" },
        scrollNudgerStyle: { color: "c-primary-sf" },
        scrollIconStyle: { color: "c-primary-obj" }
      },
      outlined: {
        ...c === "default" ? {
          triggerStyle: { borderColor: "bc-primary-mid" },
          triggerTextStyle: { color: "c-primary-obj" },
          triggerIconStyle: { color: "c-primary-obj-02" }
        } : ($ = p[c]) == null ? void 0 : $.outlined,
        menuWrapperStyle: {
          borderColor: "bc-primary-mid",
          color: "c-primary-sf"
        },
        menuStyle: { color: "c-primary-sf" },
        optionStyle: { color: "c-primary-sf" },
        optionTextStyle: { color: "c-primary-obj" },
        optionIconStyle: { color: "c-primary-obj" },
        scrollNudgerStyle: { color: "c-primary-sf" },
        scrollIconStyle: { color: "c-primary-obj" }
      }
    },
    secondary: {
      filled: {
        ...c === "default" ? {
          triggerStyle: { color: "c-secondary-sf-01" },
          triggerTextStyle: { color: "c-secondary-obj" },
          triggerIconStyle: { color: "c-secondary-obj-02" }
        } : (K = p[c]) == null ? void 0 : K.filled,
        menuWrapperStyle: {
          borderColor: "bc-secondary-mid",
          color: "c-secondary-sf"
        },
        menuStyle: { color: "c-secondary-sf" },
        optionStyle: { color: "c-secondary-sf" },
        optionTextStyle: { color: "c-secondary-obj" },
        optionIconStyle: { color: "c-secondary-obj" },
        scrollNudgerStyle: { color: "c-secondary-sf" },
        scrollIconStyle: { color: "c-secondary-obj" }
      },
      outlined: {
        ...c === "default" ? {
          triggerStyle: { borderColor: "bc-secondary-mid" },
          triggerTextStyle: { color: "c-secondary-obj" },
          triggerIconStyle: { color: "c-secondary-obj-02" }
        } : (q = p[c]) == null ? void 0 : q.outlined,
        menuWrapperStyle: {
          borderColor: "bc-secondary-mid",
          color: "c-secondary-sf"
        },
        menuStyle: { color: "c-secondary-sf" },
        optionStyle: { color: "c-secondary-sf" },
        optionTextStyle: { color: "c-secondary-obj" },
        optionIconStyle: { color: "c-secondary-obj" },
        scrollNudgerStyle: { color: "c-secondary-sf" },
        scrollIconStyle: { color: "c-secondary-obj" }
      }
    },
    tertiary: {
      filled: {
        ...c === "default" ? {
          triggerStyle: { color: "c-tertiary-sf-01" },
          triggerTextStyle: { color: "c-tertiary-obj" },
          triggerIconStyle: { color: "c-tertiary-obj-02" }
        } : (G = p[c]) == null ? void 0 : G.filled,
        menuWrapperStyle: {
          borderColor: "bc-tertiary-mid",
          color: "c-tertiary-sf"
        },
        menuStyle: { color: "c-tertiary-sf" },
        optionStyle: { color: "c-tertiary-sf" },
        optionTextStyle: { color: "c-tertiary-obj" },
        optionIconStyle: { color: "c-tertiary-obj" },
        scrollNudgerStyle: { color: "c-tertiary-sf" },
        scrollIconStyle: { color: "c-tertiary-obj" }
      },
      outlined: {
        ...c === "default" ? {
          triggerStyle: { borderColor: "bc-tertiary-mid" },
          triggerTextStyle: { color: "c-tertiary-obj" },
          triggerIconStyle: { color: "c-tertiary-obj-02" }
        } : (Q = p[c]) == null ? void 0 : Q.outlined,
        menuWrapperStyle: {
          borderColor: "bc-tertiary-mid",
          color: "c-tertiary-sf"
        },
        menuStyle: { color: "c-tertiary-sf" },
        optionStyle: { color: "c-tertiary-sf" },
        optionTextStyle: { color: "c-tertiary-obj" },
        optionIconStyle: { color: "c-tertiary-obj" },
        scrollNudgerStyle: { color: "c-tertiary-sf" },
        scrollIconStyle: { color: "c-tertiary-obj" }
      }
    },
    neutral: {
      filled: {
        ...c === "default" ? {
          triggerStyle: { color: "c-neutral-sf-01" },
          triggerTextStyle: { color: "c-neutral-obj" },
          triggerIconStyle: { color: "c-neutral-obj-02" }
        } : (Y = p[c]) == null ? void 0 : Y.filled,
        menuWrapperStyle: {
          borderColor: "bc-neutral-mid",
          color: "c-neutral-sf"
        },
        menuStyle: {
          color: "c-neutral-sf"
        },
        optionStyle: { color: "c-neutral-sf" },
        optionTextStyle: { color: "c-neutral-obj" },
        optionIconStyle: { color: "c-neutral-obj" },
        scrollNudgerStyle: { color: "c-neutral-sf" },
        scrollIconStyle: { color: "c-neutral-obj" }
      },
      outlined: {
        ...c === "default" ? {
          triggerStyle: { borderColor: "bc-neutral-mid" },
          triggerTextStyle: { color: "c-neutral-obj" },
          triggerIconStyle: { color: "c-neutral-obj-02" }
        } : (Z = p[c]) == null ? void 0 : Z.outlined,
        menuWrapperStyle: {
          borderColor: "bc-neutral-mid",
          color: "c-neutral-sf"
        },
        menuStyle: {
          color: "c-neutral-sf"
        },
        optionStyle: { color: "c-neutral-sf" },
        optionTextStyle: { color: "c-neutral-obj" },
        optionIconStyle: { color: "c-neutral-obj" },
        scrollNudgerStyle: { color: "c-neutral-sf" },
        scrollIconStyle: { color: "c-neutral-obj" }
      }
    }
  }, B = {
    outlined: {
      triggerStyle: {
        borderWidth: "bw-0.125",
        borderStyle: "bs-solid"
      }
    },
    filled: {}
  }, Br = st, Er = {
    position: "relative",
    width: (ee = (_ = o[t]) == null ? void 0 : _.rootStyle) == null ? void 0 : ee.width,
    opacity: j ? "oc-0.5" : "oc-1"
  }, Ar = at, Lr = {
    position: "relative",
    width: "w-full",
    height: (te = (re = o[t]) == null ? void 0 : re.triggerStyle) == null ? void 0 : te.height,
    paddingX: (le = (oe = o[t]) == null ? void 0 : oe.triggerStyle) == null ? void 0 : le.paddingX,
    gap: (ne = (ce = o[t]) == null ? void 0 : ce.triggerStyle) == null ? void 0 : ne.gap,
    wrap: "flex-nowrap",
    cursor: j ? "cs-not-allowed" : "cs-pointer",
    borderStyle: (se = (ie = B[i]) == null ? void 0 : ie.triggerStyle) == null ? void 0 : se.borderStyle,
    borderWidth: (de = (ye = B[i]) == null ? void 0 : ye.triggerStyle) == null ? void 0 : de.borderWidth,
    borderColor: (ue = (ge = (ae = g[a]) == null ? void 0 : ae[i]) == null ? void 0 : ge.triggerStyle) == null ? void 0 : ue.borderColor,
    color: (fe = (pe = (Se = g[a]) == null ? void 0 : Se[i]) == null ? void 0 : pe.triggerStyle) == null ? void 0 : fe.color,
    borderRadius: "rad-0.5",
    justifyContent: "jc-between",
    alignItems: "ai-center"
  }, Ur = br, Pr = {
    color: (he = (be = (me = g[a]) == null ? void 0 : me[i]) == null ? void 0 : be.triggerTextStyle) == null ? void 0 : he.color,
    fontFamily: "ff-primary",
    font: (xe = (Ie = o[t]) == null ? void 0 : Ie.triggerTextStyle) == null ? void 0 : xe.font,
    lineClamp: "lc-1",
    // 선택된 값 없으면 placeholder 표시
    opacity: n != null && n.length || I.length ? "oc-1" : "oc-0.5",
    wordBreak: "break-words"
  }, Fr = hr, Jr = {
    color: (Te = (je = (we = g[a]) == null ? void 0 : we[i]) == null ? void 0 : je.triggerIconStyle) == null ? void 0 : Te.color,
    size: (We = (ve = o[t]) == null ? void 0 : ve.triggerIconStyle) == null ? void 0 : We.size,
    minWidth: (He = (De = o[t]) == null ? void 0 : De.triggerIconStyle) == null ? void 0 : He.minWidth,
    minHeight: (ke = (Ne = o[t]) == null ? void 0 : Ne.triggerIconStyle) == null ? void 0 : ke.minHeight
  }, $r = Ir, Kr = {
    position: "absolute",
    top: R ? void 0 : "t-104%",
    bottom: R ? "b-104%" : void 0,
    left: "l-0",
    width: "w-full",
    color: (Oe = (ze = (Ce = g[a]) == null ? void 0 : Ce[i]) == null ? void 0 : ze.menuWrapperStyle) == null ? void 0 : Oe.color,
    borderWidth: "bw-0.125",
    borderStyle: "bs-solid",
    borderColor: (Me = (Ve = (Re = g[a]) == null ? void 0 : Re[i]) == null ? void 0 : Ve.menuWrapperStyle) == null ? void 0 : Me.borderColor,
    padding: (Be = (Xe = o[t]) == null ? void 0 : Xe.menuWrapperStyle) == null ? void 0 : Be.padding,
    borderRadius: "rad-0.5",
    visibility: x ? "visible" : "hidden",
    zIndex: "z-2"
  }, qr = dt, Gr = {
    position: "relative",
    width: "w-full",
    maxHeight: (Ae = (Ee = o[t]) == null ? void 0 : Ee.menuStyle) == null ? void 0 : Ae.maxHeight,
    direction: "flex-col",
    gap: (Ue = (Le = o[t]) == null ? void 0 : Le.menuStyle) == null ? void 0 : Ue.gap,
    zIndex: "z-2",
    overflow: "ovf-auto",
    scrollBar: "scr-hidden"
  }, Qr = yt, Yr = {
    position: "relative",
    width: "w-full",
    height: (Fe = (Pe = o[t]) == null ? void 0 : Pe.optionStyle) == null ? void 0 : Fe.height,
    paddingX: ($e = (Je = o[t]) == null ? void 0 : Je.optionStyle) == null ? void 0 : $e.paddingX,
    gap: (qe = (Ke = o[t]) == null ? void 0 : Ke.optionStyle) == null ? void 0 : qe.gap,
    color: (Ye = (Qe = (Ge = g[a]) == null ? void 0 : Ge[i]) == null ? void 0 : Qe.optionStyle) == null ? void 0 : Ye.color,
    alignItems: "ai-center",
    justifyContent: "jc-between",
    wrap: "flex-nowrap",
    shrink: "shrink-0",
    borderRadius: "rad-0"
  }, Zr = br, _r = {
    color: (er = (_e = (Ze = g[a]) == null ? void 0 : Ze[i]) == null ? void 0 : _e.optionTextStyle) == null ? void 0 : er.color,
    fontFamily: "ff-primary",
    font: (tr = (rr = o[t]) == null ? void 0 : rr.optionTextStyle) == null ? void 0 : tr.font,
    lineClamp: "lc-1",
    wordBreak: "break-words"
  }, et = gt, rt = {
    color: (cr = (lr = (or = g[a]) == null ? void 0 : or[i]) == null ? void 0 : lr.optionIconStyle) == null ? void 0 : cr.color,
    size: (ir = (nr = o[t]) == null ? void 0 : nr.optionIconStyle) == null ? void 0 : ir.size
  }, E = Ir, A = {
    position: "absolute",
    width: "w-full",
    padding: "p-0.5",
    color: (dr = (yr = (sr = g[a]) == null ? void 0 : sr[i]) == null ? void 0 : yr.scrollNudgerStyle) == null ? void 0 : dr.color,
    borderRadius: "rad-0.5",
    zIndex: "z-3"
  }, L = hr, U = {
    marginX: "mx-auto",
    size: (gr = (ar = o[t]) == null ? void 0 : ar.scrollIconStyle) == null ? void 0 : gr.size,
    color: (pr = (Sr = (ur = g[a]) == null ? void 0 : ur[i]) == null ? void 0 : Sr.scrollIconStyle) == null ? void 0 : pr.color
  }, v = (e) => {
    if (typeof e == "object") {
      if ("label" in e && typeof e.label == "string")
        return e.label;
      {
        let r = JSON.stringify(e);
        return r = r.replace(/^{/, "").replace(/}$/, ""), r = r.replace(/"/g, ""), r;
      }
    }
    return e;
  }, W = (e) => typeof e == "object" && "value" in e ? e.value : e, tt = (e) => {
    if (typeof e == "object")
      if ("value" in e) {
        if (typeof e.value == "string" || typeof e.value == "number")
          return String(e.value);
        {
          let r = JSON.stringify(e.value);
          return r = r.replace(/^{/, "").replace(/}$/, ""), r = r.replace(/"/g, ""), r;
        }
      } else {
        let r = JSON.stringify(e);
        return r = r.replace(/^{/, "").replace(/}$/, ""), r = r.replace(/"/g, ""), r;
      }
    return String(e);
  }, ot = (e) => {
    const r = W(e);
    return (n !== void 0 ? n : I).some((l) => l === r);
  }, lt = (e) => {
    const r = n !== void 0 ? n : I, s = W(e);
    if (r.includes(s)) {
      const l = r.filter((f) => f !== s);
      n !== void 0 || C(l), d == null || d(l);
    } else {
      const l = [...r, s];
      if (D && l.length > D) return;
      n !== void 0 || C(l), d == null || d(l);
    }
  };
  mr(() => {
    const e = (r) => {
      S.current && r.target instanceof Node && !S.current.contains(r.target) && T(!1);
    };
    return document.addEventListener("mousedown", e), () => {
      document.removeEventListener("mousedown", e);
    };
  }, []), mr(() => {
    if (x && S.current && (S.current.focus(), ct()), !h.current)
      return;
    const e = () => {
      const { scrollTop: s, scrollHeight: l, clientHeight: f } = h.current;
      Vr(s > 0), Xr(s + f < l);
    };
    e();
    const r = h.current;
    return r == null || r.addEventListener("scroll", e), () => {
      r.removeEventListener("scroll", e);
    };
  }, [x]);
  const ct = () => {
    if (!S.current) return;
    const e = S.current.getBoundingClientRect(), r = +o[t].menuStyle.maxHeight.split("-")[1] * 8, l = ut(
      S.current
    ).getBoundingClientRect(), f = e.top - l.top;
    l.bottom - e.bottom < r && f > r ? V(!0) : V(!1);
  }, P = (e) => {
    if (x && h.current) {
      const r = setInterval(() => {
        h.current.scrollBy(0, e === "up" ? -10 : 10);
      }, 50);
      O(r);
    }
  }, F = () => {
    z && (clearInterval(z), O(null));
  }, nt = () => {
    const e = n !== void 0 ? n : I;
    return e.length ? e.map((s) => {
      const l = b.find((f) => W(f) === s);
      return l ? v(l) : String(s);
    }).join(", ") : Tr || "선택해주세요.";
  }, it = (e) => {
    e.key === "ArrowDown" ? X(
      (r) => r < b.length - 1 ? r + 1 : 0
    ) : e.key === "ArrowUp" ? X(
      (r) => r > 0 ? r - 1 : b.length - 1
    ) : e.key === "Enter" ? M >= 0 && M < b.length : e.key === "Escape" && T(!1);
  };
  return /* @__PURE__ */ w(
    Br,
    {
      ...y(Er, vr || {}),
      noDefaultStyle: !0,
      ref: S,
      htmlAttributes: { tabIndex: 0, onKeyDown: it },
      children: [
        /* @__PURE__ */ u(
          "select",
          {
            className: "hidden",
            id: xr,
            name: wr,
            onChange: () => {
            },
            required: jr,
            multiple: !0,
            value: (n !== void 0 ? n : I).map(
              (e) => String(e)
            ),
            children: b.map((e, r) => /* @__PURE__ */ u("option", { value: tt(e), children: v(e) }, r))
          }
        ),
        /* @__PURE__ */ w(
          Ar,
          {
            ...y(Lr, Wr || {}),
            noDefaultStyle: !0,
            htmlAttributes: {
              onClick: j ? void 0 : () => T((e) => !e)
            },
            children: [
              /* @__PURE__ */ u(
                Ur,
                {
                  ...y(Pr, Dr || {}),
                  noDefaultStyle: !0,
                  children: nt()
                }
              ),
              /* @__PURE__ */ u(
                Fr,
                {
                  type: x ? "up" : "down",
                  ...y(Jr, Hr || {}),
                  noDefaultStyle: !0
                }
              )
            ]
          }
        ),
        /* @__PURE__ */ w(
          $r,
          {
            ...y(Kr, Nr || {}),
            noDefaultStyle: !0,
            children: [
              /* @__PURE__ */ u(
                E,
                {
                  top: "t-0",
                  left: "l-0",
                  visibility: Rr ? "visible" : "hidden",
                  ...y(A, N || {}),
                  htmlAttributes: {
                    onMouseEnter: () => P("up"),
                    onMouseLeave: F
                  },
                  noDefaultStyle: !0,
                  children: /* @__PURE__ */ u(
                    L,
                    {
                      type: "up",
                      ...y(U, k || {}),
                      noDefaultStyle: !0
                    }
                  )
                }
              ),
              /* @__PURE__ */ u(
                qr,
                {
                  ref: h,
                  ...y(Gr, kr || {}),
                  noDefaultStyle: !0,
                  children: b.map((e, r) => {
                    const s = ot(e);
                    return /* @__PURE__ */ w(
                      Qr,
                      {
                        onClick: () => lt(e),
                        ...y(Yr, Cr || {}),
                        noDefaultStyle: !0,
                        hover: !0,
                        children: [
                          /* @__PURE__ */ u(
                            Zr,
                            {
                              ...y(_r, zr || {}),
                              noDefaultStyle: !0,
                              children: v(e)
                            }
                          ),
                          /* @__PURE__ */ u(
                            et,
                            {
                              visibility: s ? "visible" : "hidden",
                              ...y(rt, Or || {})
                            }
                          )
                        ]
                      },
                      r
                    );
                  })
                }
              ),
              /* @__PURE__ */ u(
                E,
                {
                  bottom: "b-0",
                  left: "l-0",
                  visibility: Mr ? "visible" : "hidden",
                  ...y(A, N || {}),
                  htmlAttributes: {
                    onMouseEnter: () => P("down"),
                    onMouseLeave: F
                  },
                  noDefaultStyle: !0,
                  children: /* @__PURE__ */ u(
                    L,
                    {
                      type: "down",
                      ...y(U, k || {}),
                      noDefaultStyle: !0
                    }
                  )
                }
              )
            ]
          }
        )
      ]
    }
  );
};
export {
  Dt as default
};
