// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Molecules/NumberInput.js
"use client";
import { jsxs as u, jsx as l } from "react/jsx-runtime";
import x from "../Icons/IconMinus.js";
import B from "../Icons/IconPlus.js";
import { useState as y } from "react";
import R from "../Atoms/Input.js";
import m from "../Atoms/Flex.js";
import X from "../Atoms/Text.js";
import g from "../Atoms/Button.js";
const E = ({
  mainStyle: s,
  inputWrapperStyle: b,
  inputStyle: p,
  buttonContainerStyle: N,
  buttonStyle: a,
  amountStyle: w,
  value: t,
  defaultValue: z,
  min: h,
  max: i,
  step: o = 1,
  hasButton: I = !0,
  hasAmount: M = !0,
  setValue: n
}) => {
  const [e, c] = y(
    String(t || z || 0)
  ), S = (r) => {
    const f = r.target.value;
    if (f === "")
      c("");
    else if (/^\d*$/.test(f)) {
      let d = Number(f);
      i && (d = Math.min(d, i)), h && (d = Math.max(d, h)), n && n(d), c(String(d));
    }
  }, j = () => {
    e === "" && (n && n(h || 0), c(String(h || 0)));
  }, C = () => {
    if (t) {
      const r = Number(t) + o;
      n && n(r), c(String(r));
    } else
      n && n(
        i ? Math.min(Number(e) + o, i) : Number(e) + o
      ), c(
        (r) => String(i ? Math.min(Number(r) + o, i) : Number(r) + o)
      );
  }, k = () => {
    if (t) {
      const r = Number(t) - o;
      n && n(r), c(String(r));
    } else
      n && n(
        i ? Math.min(Number(e) - o, i) : Number(e) - o
      ), c(
        (r) => String(i ? Math.min(Number(r) - o, i) : Number(r) - o)
      );
  };
  return /* @__PURE__ */ u(
    m,
    {
      width: "w-10",
      wrap: "flex-nowrap",
      color: "c-neutral-sf-03",
      padding: "p-0.5",
      borderRadius: "rad-0.5",
      ...s,
      children: [
        /* @__PURE__ */ l(
          m,
          {
            grow: "grow-1",
            justifyContent: "jc-center",
            color: "c-neutral-sf-02",
            borderRadius: "rad-0.5",
            ...b,
            children: /* @__PURE__ */ l(
              R,
              {
                type: "text",
                value: e === "" || !t ? e : t,
                width: "w-full",
                paddingX: "px-0.25",
                paddingY: "py-0.125",
                textAlign: "ta-center",
                onChange: S,
                onBlur: j,
                ...p
              }
            )
          }
        ),
        M && /* @__PURE__ */ l(m, { shrink: "shrink-0", alignItems: "ai-center", children: /* @__PURE__ */ u(
          X,
          {
            color: "c-neutral-obj-03",
            font: "f-2",
            marginX: "mx-0.75",
            ...w,
            children: [
              "/ ",
              i
            ]
          }
        ) }),
        I && /* @__PURE__ */ u(m, { direction: "flex-col", ...N, children: [
          /* @__PURE__ */ l(
            g,
            {
              padding: "p-0",
              size: "sz-1.5",
              onClick: C,
              ...a,
              children: /* @__PURE__ */ l(B, { size: "sz-full", color: "c-inherit" })
            }
          ),
          /* @__PURE__ */ l(
            g,
            {
              padding: "p-0",
              size: "sz-1.5",
              onClick: k,
              ...a,
              children: /* @__PURE__ */ l(x, { size: "sz-full", color: "c-inherit" })
            }
          )
        ] })
      ]
    }
  );
};
export {
  E as default
};
