// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Molecules/SegmentedControl.js
"use client";
import { jsx as o, jsxs as a } from "react/jsx-runtime";
import g from "../Icons/IconCheck.js";
import l from "../Atoms/Flex.js";
import n from "../Atoms/Text.js";
import b from "../Atoms/Button.js";
const R = ({
  segments: d,
  currentSegment: c,
  handleSegmentChange: s,
  borderRadius: e
}) => {
  const h = Object.keys(d), t = Object.values(d);
  return /* @__PURE__ */ o(
    l,
    {
      width: "w-full",
      borderRadius: e,
      padding: "p-0",
      color: "c-transparent",
      borderColor: "bc-neutral-mid",
      borderStyle: "bs-solid",
      borderWidth: "bw-0.25",
      children: t.map((i, r) => /* @__PURE__ */ o(l, { grow: "grow-1", shrink: "shrink-1", children: h[r] === c ? /* @__PURE__ */ a(
        l,
        {
          width: "w-full",
          color: "c-primary-sf",
          borderRadius: e,
          ...r !== 0 && {
            borderRadiusLeft: "rad-l-0"
          },
          ...r !== t.length - 1 && {
            borderRadiusRight: "rad-r-0"
          },
          paddingX: "px-0",
          paddingY: "py-1",
          borderRightColor: "brc-neutral-mid",
          borderRightStyle: "brs-solid",
          borderRightWidth: r === t.length - 1 ? "brw-0" : "brw-0.25",
          gap: "g-1",
          children: [
            /* @__PURE__ */ o(g, { size: "sz-[14px]", color: "c-inherit" }),
            /* @__PURE__ */ o(n, { font: "f-1.75", color: "c-inherit", children: i })
          ]
        }
      ) : /* @__PURE__ */ o(
        b,
        {
          width: "w-full",
          color: "c-neutral-sf",
          hoverStyle: { color: "hover:c-neutral-sf-02" },
          borderRadius: e,
          ...r !== 0 && {
            borderRadiusLeft: "rad-l-0"
          },
          ...r !== t.length - 1 && {
            borderRadiusRight: "rad-r-0"
          },
          paddingX: "px-0",
          paddingY: "py-1",
          borderRightColor: "brc-neutral-mid",
          borderRightStyle: "brs-solid",
          borderRightWidth: r === t.length - 1 ? "brw-0" : "brw-0.25",
          onClick: () => s(h[r]),
          children: /* @__PURE__ */ o(n, { font: "f-1.75", color: "c-inherit", children: i })
        }
      ) }, i))
    }
  );
};
export {
  R as default
};
