// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Molecules/Select.js
"use client";
import { jsxs as m, jsx as y } from "react/jsx-runtime";
import { useState as f, useRef as ur, useEffect as Sr } from "react";
import oo from "../Atoms/WrapperWithRef.js";
import to from "../Atoms/Button.js";
import fr from "../Atoms/Text.js";
import lo from "../Atoms/ScrollWithRef.js";
import pr from "../Icons/IconArrow.js";
import co from "../Atoms/Flex.js";
import n from "../../utils/deepMerge.js";
import no from "../Icons/IconCheck.js";
import br from "../Atoms/Wrapper.js";
import { findScrollableOrHiddenParent as io } from "../../utils/client/index.js";
const jo = ({
  id: mr,
  name: hr,
  required: xr,
  options: S,
  placeholder: W,
  value: s,
  defaultValue: Ir,
  size: o = "md",
  color: i = "neutral",
  state: t = "default",
  theme: c = "outlined",
  disabled: I,
  onChange: D,
  rootStyle: jr,
  triggerStyle: wr,
  triggerTextStyle: Tr,
  triggerIconStyle: vr,
  menuWrapperStyle: Cr,
  menuStyle: Wr,
  optionStyle: Dr,
  optionTextStyle: Nr,
  optionIconStyle: zr,
  scrollNudgerStyle: N,
  scrollIconStyle: z
}) => {
  var $, K, q, G, Q, Y, Z, _, ee, re, oe, te, le, ce, ne, ie, ae, ye, se, de, ge, ue, Se, fe, pe, be, me, he, xe, Ie, je, we, Te, ve, Ce, We, De, Ne, ze, ke, He, Re, Oe, Me, Xe, Be, Ee, Ae, Le, Ve, Ue, Pe, Fe, Je, $e, Ke, qe, Ge, Qe, Ye, Ze, _e, er, rr, or, tr, lr, cr, nr, ir, ar, yr, sr, dr, gr;
  const [b, h] = f(!1), d = ur(null), p = ur(null), [g, kr] = f(
    s || Ir || null
  ), [Hr, Rr] = f(!1), [Or, Mr] = f(!1), [k, H] = f(
    null
  ), [R, O] = f(!1), [M, X] = f(-1), l = {
    sm: {
      rootStyle: { width: "w-24" },
      triggerStyle: { height: "h-2.75", paddingX: "px-1", gap: "g-1" },
      triggerTextStyle: { font: "f-1.5" },
      triggerIconStyle: { size: "sz-1.75" },
      menuWrapperStyle: { padding: "p-0.5" },
      menuStyle: { maxHeight: "xh-22", gap: "g-0.5" },
      optionStyle: { height: "h-2.75", paddingX: "px-1", gap: "g-1" },
      optionTextStyle: { font: "f-1.5" },
      optionIconStyle: { size: "sz-1.75" },
      scrollIconStyle: { size: "sz-1.75" }
    },
    md: {
      rootStyle: { width: "w-32" },
      triggerStyle: { height: "h-4.25", paddingX: "px-1.5", gap: "g-1.5" },
      triggerTextStyle: { font: "f-1.75" },
      triggerIconStyle: { size: "sz-2.25" },
      menuWrapperStyle: { padding: "p-0.5" },
      menuStyle: { maxHeight: "xh-30", gap: "g-0.5" },
      optionStyle: { height: "h-4.25", paddingX: "px-1.5", gap: "g-1.5" },
      optionTextStyle: { font: "f-1.75" },
      optionIconStyle: { size: "sz-2.25" },
      scrollIconStyle: { size: "sz-2.25" }
    },
    lg: {
      rootStyle: { width: "w-40" },
      triggerStyle: { height: "h-4.5", paddingX: "px-1.5", gap: "g-1.5" },
      triggerTextStyle: { font: "f-2" },
      triggerIconStyle: { size: "sz-2.5" },
      menuWrapperStyle: { padding: "p-0.5" },
      menuStyle: { maxHeight: "xh-36", gap: "g-0.5" },
      optionStyle: { height: "h-4.5", paddingX: "px-1.5", gap: "g-1.5" },
      optionTextStyle: { font: "f-2" },
      optionIconStyle: { size: "sz-2.5" },
      scrollIconStyle: { size: "sz-2.5" }
    }
  }, u = {
    error: {
      filled: {
        triggerStyle: { color: "c-error-sf-01" },
        triggerTextStyle: { color: "c-error-obj-01" },
        triggerIconStyle: { color: "c-error-sf-04" }
      },
      outlined: {
        triggerStyle: { borderColor: "bc-error-mid" },
        triggerTextStyle: { color: "c-error-mid" },
        triggerIconStyle: { color: "c-error-obj-01" }
      }
    },
    warning: {
      filled: {
        triggerStyle: { color: "c-warning-sf-01" },
        triggerTextStyle: { color: "c-warning-obj-01" },
        triggerIconStyle: { color: "c-warning-sf-04" }
      },
      outlined: {
        triggerStyle: { borderColor: "bc-warning-mid" },
        triggerTextStyle: { color: "c-warning-mid" },
        triggerIconStyle: { color: "c-warning-obj-01" }
      }
    },
    success: {
      filled: {
        triggerStyle: { color: "c-success-sf-01" },
        triggerTextStyle: { color: "c-success-obj-01" },
        triggerIconStyle: { color: "c-success-sf-04" }
      },
      outlined: {
        triggerStyle: { borderColor: "bc-success-mid" },
        triggerTextStyle: { color: "c-success-mid" },
        triggerIconStyle: { color: "c-success-obj-01" }
      }
    },
    perfect: {
      filled: {
        triggerStyle: { color: "c-perfect-sf-01" },
        triggerTextStyle: { color: "c-perfect-obj-01" },
        triggerIconStyle: { color: "c-perfect-sf-04" }
      },
      outlined: {
        triggerStyle: { borderColor: "bc-perfect-mid" },
        triggerTextStyle: { color: "c-perfect-mid" },
        triggerIconStyle: { color: "c-perfect-obj-01" }
      }
    }
  }, a = {
    primary: {
      filled: {
        ...t === "default" ? {
          triggerStyle: { color: "c-primary-sf-01" },
          triggerTextStyle: { color: "c-primary-obj" },
          triggerIconStyle: { color: "c-primary-obj-02" }
        } : ($ = u[t]) == null ? void 0 : $.filled,
        menuWrapperStyle: {
          borderColor: "bc-primary-mid",
          color: "c-primary-sf"
        },
        menuStyle: { color: "c-primary-sf" },
        optionStyle: { color: "c-primary-sf" },
        optionTextStyle: { color: "c-primary-obj" },
        optionIconStyle: { color: "c-primary-obj" },
        scrollNudgerStyle: { color: "c-primary-sf" },
        scrollIconStyle: { color: "c-primary-obj" }
      },
      outlined: {
        ...t === "default" ? {
          triggerStyle: { borderColor: "bc-primary-mid" },
          triggerTextStyle: { color: "c-primary-obj" },
          triggerIconStyle: { color: "c-primary-obj-02" }
        } : (K = u[t]) == null ? void 0 : K.outlined,
        menuWrapperStyle: {
          borderColor: "bc-primary-mid",
          color: "c-primary-sf"
        },
        menuStyle: { color: "c-primary-sf" },
        optionStyle: { color: "c-primary-sf" },
        optionTextStyle: { color: "c-primary-obj" },
        optionIconStyle: { color: "c-primary-obj" },
        scrollNudgerStyle: { color: "c-primary-sf" },
        scrollIconStyle: { color: "c-primary-obj" }
      }
    },
    secondary: {
      filled: {
        ...t === "default" ? {
          triggerStyle: { color: "c-secondary-sf-01" },
          triggerTextStyle: { color: "c-secondary-obj" },
          triggerIconStyle: { color: "c-secondary-obj-02" }
        } : (q = u[t]) == null ? void 0 : q.filled,
        menuWrapperStyle: {
          borderColor: "bc-secondary-mid",
          color: "c-secondary-sf"
        },
        menuStyle: { color: "c-secondary-sf" },
        optionStyle: { color: "c-secondary-sf" },
        optionTextStyle: { color: "c-secondary-obj" },
        optionIconStyle: { color: "c-secondary-obj" },
        scrollNudgerStyle: { color: "c-secondary-sf" },
        scrollIconStyle: { color: "c-secondary-obj" }
      },
      outlined: {
        ...t === "default" ? {
          triggerStyle: { borderColor: "bc-secondary-mid" },
          triggerTextStyle: { color: "c-secondary-obj" },
          triggerIconStyle: { color: "c-secondary-obj-02" }
        } : (G = u[t]) == null ? void 0 : G.outlined,
        menuWrapperStyle: {
          borderColor: "bc-secondary-mid",
          color: "c-secondary-sf"
        },
        menuStyle: { color: "c-secondary-sf" },
        optionStyle: { color: "c-secondary-sf" },
        optionTextStyle: { color: "c-secondary-obj" },
        optionIconStyle: { color: "c-secondary-obj" },
        scrollNudgerStyle: { color: "c-secondary-sf" },
        scrollIconStyle: { color: "c-secondary-obj" }
      }
    },
    tertiary: {
      filled: {
        ...t === "default" ? {
          triggerStyle: { color: "c-tertiary-sf-01" },
          triggerTextStyle: { color: "c-tertiary-obj" },
          triggerIconStyle: { color: "c-tertiary-obj-02" }
        } : (Q = u[t]) == null ? void 0 : Q.filled,
        menuWrapperStyle: {
          borderColor: "bc-tertiary-mid",
          color: "c-tertiary-sf"
        },
        menuStyle: { color: "c-tertiary-sf" },
        optionStyle: { color: "c-tertiary-sf" },
        optionTextStyle: { color: "c-tertiary-obj" },
        optionIconStyle: { color: "c-tertiary-obj" },
        scrollNudgerStyle: { color: "c-tertiary-sf" },
        scrollIconStyle: { color: "c-tertiary-obj" }
      },
      outlined: {
        ...t === "default" ? {
          triggerStyle: { borderColor: "bc-tertiary-mid" },
          triggerTextStyle: { color: "c-tertiary-obj" },
          triggerIconStyle: { color: "c-tertiary-obj-02" }
        } : (Y = u[t]) == null ? void 0 : Y.outlined,
        menuWrapperStyle: {
          borderColor: "bc-tertiary-mid",
          color: "c-tertiary-sf"
        },
        menuStyle: { color: "c-tertiary-sf" },
        optionStyle: { color: "c-tertiary-sf" },
        optionTextStyle: { color: "c-tertiary-obj" },
        optionIconStyle: { color: "c-tertiary-obj" },
        scrollNudgerStyle: { color: "c-tertiary-sf" },
        scrollIconStyle: { color: "c-tertiary-obj" }
      }
    },
    neutral: {
      filled: {
        ...t === "default" ? {
          triggerStyle: { color: "c-neutral-sf-01" },
          triggerTextStyle: { color: "c-neutral-obj" },
          triggerIconStyle: { color: "c-neutral-obj-02" }
        } : (Z = u[t]) == null ? void 0 : Z.filled,
        menuWrapperStyle: {
          borderColor: "bc-neutral-mid",
          color: "c-neutral-sf"
        },
        menuStyle: {
          color: "c-neutral-sf"
        },
        optionStyle: { color: "c-neutral-sf" },
        optionTextStyle: { color: "c-neutral-obj" },
        optionIconStyle: { color: "c-neutral-obj" },
        scrollNudgerStyle: { color: "c-neutral-sf" },
        scrollIconStyle: { color: "c-neutral-obj" }
      },
      outlined: {
        ...t === "default" ? {
          triggerStyle: { borderColor: "bc-neutral-mid" },
          triggerTextStyle: { color: "c-neutral-obj" },
          triggerIconStyle: { color: "c-neutral-obj-02" }
        } : (_ = u[t]) == null ? void 0 : _.outlined,
        menuWrapperStyle: {
          borderColor: "bc-neutral-mid",
          color: "c-neutral-sf"
        },
        menuStyle: {
          color: "c-neutral-sf"
        },
        optionStyle: { color: "c-neutral-sf" },
        optionTextStyle: { color: "c-neutral-obj" },
        optionIconStyle: { color: "c-neutral-obj" },
        scrollNudgerStyle: { color: "c-neutral-sf" },
        scrollIconStyle: { color: "c-neutral-obj" }
      }
    }
  }, B = {
    outlined: {
      triggerStyle: {
        borderWidth: "bw-0.125",
        borderStyle: "bs-solid"
      }
    },
    filled: {}
  }, Xr = oo, Br = {
    position: "relative",
    width: (re = (ee = l[o]) == null ? void 0 : ee.rootStyle) == null ? void 0 : re.width,
    opacity: I ? "oc-0.5" : "oc-1"
  }, Er = co, Ar = {
    position: "relative",
    width: "w-full",
    height: (te = (oe = l[o]) == null ? void 0 : oe.triggerStyle) == null ? void 0 : te.height,
    paddingX: (ce = (le = l[o]) == null ? void 0 : le.triggerStyle) == null ? void 0 : ce.paddingX,
    gap: (ie = (ne = l[o]) == null ? void 0 : ne.triggerStyle) == null ? void 0 : ie.gap,
    wrap: "flex-nowrap",
    cursor: I ? "cs-not-allowed" : "cs-pointer",
    borderStyle: (ye = (ae = B[c]) == null ? void 0 : ae.triggerStyle) == null ? void 0 : ye.borderStyle,
    borderWidth: (de = (se = B[c]) == null ? void 0 : se.triggerStyle) == null ? void 0 : de.borderWidth,
    borderColor: (Se = (ue = (ge = a[i]) == null ? void 0 : ge[c]) == null ? void 0 : ue.triggerStyle) == null ? void 0 : Se.borderColor,
    color: (be = (pe = (fe = a[i]) == null ? void 0 : fe[c]) == null ? void 0 : pe.triggerStyle) == null ? void 0 : be.color,
    borderRadius: "rad-0.5",
    justifyContent: "jc-between",
    alignItems: "ai-center"
  }, Lr = fr, Vr = {
    color: (xe = (he = (me = a[i]) == null ? void 0 : me[c]) == null ? void 0 : he.triggerTextStyle) == null ? void 0 : xe.color,
    fontFamily: "ff-primary",
    font: (je = (Ie = l[o]) == null ? void 0 : Ie.triggerTextStyle) == null ? void 0 : je.font,
    lineClamp: "lc-1",
    opacity: s || g ? "oc-1" : "oc-0.5",
    wordBreak: "break-words"
  }, Ur = pr, Pr = {
    color: (ve = (Te = (we = a[i]) == null ? void 0 : we[c]) == null ? void 0 : Te.triggerIconStyle) == null ? void 0 : ve.color,
    size: (We = (Ce = l[o]) == null ? void 0 : Ce.triggerIconStyle) == null ? void 0 : We.size
  }, Fr = br, Jr = {
    position: "absolute",
    top: R ? void 0 : "t-104%",
    bottom: R ? "b-104%" : void 0,
    left: "l-0",
    width: "w-full",
    color: (ze = (Ne = (De = a[i]) == null ? void 0 : De[c]) == null ? void 0 : Ne.menuWrapperStyle) == null ? void 0 : ze.color,
    borderWidth: "bw-0.125",
    borderStyle: "bs-solid",
    borderColor: (Re = (He = (ke = a[i]) == null ? void 0 : ke[c]) == null ? void 0 : He.menuWrapperStyle) == null ? void 0 : Re.borderColor,
    padding: (Me = (Oe = l[o]) == null ? void 0 : Oe.menuWrapperStyle) == null ? void 0 : Me.padding,
    borderRadius: "rad-0.5",
    visibility: b ? "visible" : "hidden",
    zIndex: "z-2"
  }, $r = lo, Kr = {
    position: "relative",
    width: "w-full",
    maxHeight: (Be = (Xe = l[o]) == null ? void 0 : Xe.menuStyle) == null ? void 0 : Be.maxHeight,
    direction: "flex-col",
    gap: (Ae = (Ee = l[o]) == null ? void 0 : Ee.menuStyle) == null ? void 0 : Ae.gap,
    // color: colorVariants[color]?.[theme]?.menuStyle?.color,
    zIndex: "z-2",
    overflow: "ovf-auto",
    scrollBar: "scr-hidden"
  }, qr = to, Gr = {
    position: "relative",
    width: "w-full",
    height: (Ve = (Le = l[o]) == null ? void 0 : Le.optionStyle) == null ? void 0 : Ve.height,
    paddingX: (Pe = (Ue = l[o]) == null ? void 0 : Ue.optionStyle) == null ? void 0 : Pe.paddingX,
    gap: (Je = (Fe = l[o]) == null ? void 0 : Fe.optionStyle) == null ? void 0 : Je.gap,
    color: (qe = (Ke = ($e = a[i]) == null ? void 0 : $e[c]) == null ? void 0 : Ke.optionStyle) == null ? void 0 : qe.color,
    alignItems: "ai-center",
    justifyContent: "jc-between",
    wrap: "flex-nowrap",
    shrink: "shrink-0",
    borderRadius: "rad-0"
  }, Qr = fr, Yr = {
    color: (Ye = (Qe = (Ge = a[i]) == null ? void 0 : Ge[c]) == null ? void 0 : Qe.optionTextStyle) == null ? void 0 : Ye.color,
    fontFamily: "ff-primary",
    font: (_e = (Ze = l[o]) == null ? void 0 : Ze.optionTextStyle) == null ? void 0 : _e.font,
    lineClamp: "lc-1",
    wordBreak: "break-words"
  }, Zr = no, _r = {
    color: (or = (rr = (er = a[i]) == null ? void 0 : er[c]) == null ? void 0 : rr.optionIconStyle) == null ? void 0 : or.color,
    size: (lr = (tr = l[o]) == null ? void 0 : tr.optionIconStyle) == null ? void 0 : lr.size
  }, E = br, A = {
    position: "absolute",
    width: "w-full",
    padding: "p-0.5",
    color: (ir = (nr = (cr = a[i]) == null ? void 0 : cr[c]) == null ? void 0 : nr.scrollNudgerStyle) == null ? void 0 : ir.color,
    borderRadius: "rad-0.5",
    zIndex: "z-3"
  }, L = pr, V = {
    marginX: "mx-auto",
    size: (yr = (ar = l[o]) == null ? void 0 : ar.scrollIconStyle) == null ? void 0 : yr.size,
    color: (gr = (dr = (sr = a[i]) == null ? void 0 : sr[c]) == null ? void 0 : dr.scrollIconStyle) == null ? void 0 : gr.color
  }, j = (e) => {
    if (typeof e == "object") {
      if ("label" in e && typeof e.label == "string")
        return e.label;
      {
        let r = JSON.stringify(e);
        return r = r.replace(/^{/, "").replace(/}$/, ""), r = r.replace(/"/g, ""), r;
      }
    }
    return e;
  }, U = (e) => typeof e == "object" && "label" in e && "value" in e ? e.value : e, w = (e) => {
    if (typeof e == "object")
      if ("value" in e) {
        if (typeof e.value == "string" || typeof e.value == "number")
          return String(e.value);
        {
          let r = JSON.stringify(e.value);
          return r = r.replace(/^{/, "").replace(/}$/, ""), r = r.replace(/"/g, ""), r;
        }
      } else {
        let r = JSON.stringify(e);
        return r = r.replace(/^{/, "").replace(/}$/, ""), r = r.replace(/"/g, ""), r;
      }
    return String(e);
  }, T = (e) => typeof e == "object" && "value" in e && "label" in e ? s != null || g != null ? (s ?? g) === e.value : !1 : s != null || g != null ? (s ?? g) === e : !1, P = () => {
    if (d.current) {
      const e = d.current.getBoundingClientRect(), r = +l[o].menuStyle.maxHeight.split("-")[1] * 8, x = io(
        d.current
      ).getBoundingClientRect(), C = e.top - x.top;
      x.bottom - e.bottom < r && C > r ? O(!0) : O(!1);
    }
  }, F = (e) => {
    if (b && p.current) {
      const r = setInterval(() => {
        p.current.scrollBy(0, e === "up" ? -10 : 10);
      }, 50);
      H(r);
    }
  }, J = () => {
    k && (clearInterval(k), H(null));
  };
  Sr(() => {
    P();
    const e = (r) => {
      d.current && r.target instanceof Node && !d.current.contains(r.target) && h(!1);
    };
    return document.addEventListener("mousedown", e), () => {
      document.removeEventListener("mousedown", e);
    };
  }, []), Sr(() => {
    if (b && d.current && (d.current.focus(), P()), !p.current)
      return;
    const e = () => {
      const { scrollTop: v, scrollHeight: x, clientHeight: C } = p.current;
      Rr(v > 0), Mr(v + C < x - 1);
    };
    e();
    const r = p.current;
    return r == null || r.addEventListener("scroll", e), () => {
      r.removeEventListener("scroll", e);
    };
  }, [b]);
  const eo = (e) => {
    D ? D(U(e)) : kr(U(e)), h(!1);
  }, ro = (e) => {
    e.key === "ArrowDown" ? X(
      (r) => r < S.length - 1 ? r + 1 : 0
    ) : e.key === "ArrowUp" ? X(
      (r) => r > 0 ? r - 1 : S.length - 1
    ) : e.key === "Enter" ? M >= 0 && M < S.length : e.key === "Escape" && h(!1);
  };
  return /* @__PURE__ */ m(
    Xr,
    {
      ...n(Br, jr || {}),
      noDefaultStyle: !0,
      ref: d,
      htmlAttributes: { tabIndex: 0, onKeyDown: ro },
      children: [
        /* @__PURE__ */ m(
          "select",
          {
            className: "hidden",
            id: mr,
            name: hr,
            onChange: () => {
            },
            required: xr,
            value: s ? w(s) : g ? w(g) : void 0,
            children: [
              !g && /* @__PURE__ */ y("option", { value: "", disabled: !0, hidden: !0, children: W || "Select an option" }),
              S.map((e, r) => /* @__PURE__ */ y("option", { value: w(e), children: j(e) }, r))
            ]
          }
        ),
        /* @__PURE__ */ m(
          Er,
          {
            ...n(Ar, wr || {}),
            noDefaultStyle: !0,
            htmlAttributes: {
              onClick: I ? void 0 : () => h((e) => !e)
            },
            children: [
              /* @__PURE__ */ y(
                Lr,
                {
                  ...n(Vr, Tr || {}),
                  noDefaultStyle: !0,
                  children: S.some((e) => T(e)) ? j(S.find((e) => T(e))) : W || "선택해주세요."
                }
              ),
              /* @__PURE__ */ y(
                Ur,
                {
                  type: b ? "up" : "down",
                  ...n(Pr, vr || {}),
                  noDefaultStyle: !0
                }
              )
            ]
          }
        ),
        /* @__PURE__ */ m(
          Fr,
          {
            ...n(Jr, Cr || {}),
            noDefaultStyle: !0,
            children: [
              /* @__PURE__ */ y(
                E,
                {
                  top: "t-0",
                  left: "l-0",
                  visibility: Hr ? "visible" : "hidden",
                  ...n(A, N || {}),
                  htmlAttributes: {
                    onMouseEnter: () => F("up"),
                    onMouseLeave: J
                  },
                  noDefaultStyle: !0,
                  children: /* @__PURE__ */ y(
                    L,
                    {
                      type: "up",
                      ...n(V, z || {}),
                      noDefaultStyle: !0
                    }
                  )
                }
              ),
              /* @__PURE__ */ y(
                $r,
                {
                  ref: p,
                  ...n(Kr, Wr || {}),
                  noDefaultStyle: !0,
                  children: S.map((e, r) => /* @__PURE__ */ m(
                    qr,
                    {
                      onClick: () => eo(e),
                      ...n(Gr, Dr || {}),
                      noDefaultStyle: !0,
                      hover: !0,
                      children: [
                        /* @__PURE__ */ y(
                          Qr,
                          {
                            ...n(Yr, Nr || {}),
                            noDefaultStyle: !0,
                            children: j(e)
                          }
                        ),
                        /* @__PURE__ */ y(
                          Zr,
                          {
                            visibility: T(e) ? "visible" : "hidden",
                            ...n(_r, zr || {})
                          }
                        )
                      ]
                    },
                    r
                  ))
                }
              ),
              /* @__PURE__ */ y(
                E,
                {
                  bottom: "b-0",
                  left: "l-0",
                  visibility: Or ? "visible" : "hidden",
                  ...n(A, N || {}),
                  htmlAttributes: {
                    onMouseEnter: () => F("down"),
                    onMouseLeave: J
                  },
                  noDefaultStyle: !0,
                  children: /* @__PURE__ */ y(
                    L,
                    {
                      type: "down",
                      ...n(V, z || {}),
                      noDefaultStyle: !0
                    }
                  )
                }
              )
            ]
          }
        )
      ]
    }
  );
};
export {
  jo as default
};
