// @flowwwkr/design-system-tailwind@0.12.12/dist/components/Molecules/Tooltip.js
"use client";
import { jsxs as n, jsx as t } from "react/jsx-runtime";
import h from "../Atoms/Flex.js";
import u from "../Atoms/Scroll.js";
import g from "../Atoms/Text.js";
import x from "../Atoms/Wrapper.js";
import y from "../Icons/IconInfo.js";
import r from "../../utils/deepMerge.js";
const M = ({
  position: o = "top-middle",
  content: l,
  size: c = "sm",
  iconProps: a,
  wrapperColor: e,
  wrapperStyle: i,
  textStyle: m
}) => {
  const p = {
    sm: "sz-2.5",
    md: "sz-3.5",
    lg: "sz-4.5"
  }, d = {
    "right-top": "l-[calc(100%+0.6rem)] b-0",
    "right-middle": "l-[calc(100%+0.6rem)] b-50% trly-50%",
    "right-bottom": "l-[calc(100%+0.6rem)] t-0",
    "left-top": "r-[calc(100%+0.6rem)] b-0",
    "left-middle": "r-[calc(100%+0.6rem)] b-50% trly-50%",
    "left-bottom": "r-[calc(100%+0.6rem)] t-0",
    "top-right": "b-[calc(100%+0.6rem)] l-0",
    "top-middle": "b-[calc(100%+0.6rem)] l-50% -trlx-50%",
    "top-left": "b-[calc(100%+0.6rem)] r-0",
    "bottom-right": "t-[calc(100%+0.6rem)] l-0",
    "bottom-middle": "t-[calc(100%+0.6rem)] l-50% -trlx-50%",
    "bottom-left": "t-[calc(100%+0.6rem)] r-0"
  }, s = {
    "right-top": "l-[calc(100%+0.2rem)] b-50% trly-50%",
    "right-middle": "l-[calc(100%+0.2rem)] b-50% trly-50%",
    "right-bottom": "l-[calc(100%+0.2rem)] b-50% trly-50%",
    "left-top": "r-[calc(100%+0.2rem)] b-50% trly-50%",
    "left-middle": "r-[calc(100%+0.2rem)] b-50% trly-50%",
    "left-bottom": "r-[calc(100%+0.2rem)] b-50% trly-50%",
    "top-right": "b-[calc(100%+0.2rem)] l-50% -trlx-50%",
    "top-middle": "b-[calc(100%+0.2rem)] l-50% -trlx-50%",
    "top-left": "b-[calc(100%+0.2rem)] l-50% -trlx-50%",
    "bottom-right": "t-[calc(100%+0.2rem)] l-50% -trlx-50%",
    "bottom-middle": "t-[calc(100%+0.2rem)] l-50% -trlx-50%",
    "bottom-left": "t-[calc(100%+0.2rem)] l-50% -trlx-50%"
  }, b = {
    color: e || "c-neutral-obj",
    maxHeight: "xh-30",
    maxWidth: "xw-26",
    overflowX: "ovf-x-hidden",
    overflowY: "ovf-y-auto",
    display: "hidden",
    position: "absolute",
    paddingY: "py-1",
    paddingX: "px-2",
    borderRadius: "rad-1",
    scrollBar: "scr-hover",
    scrollBarSize: "sbs-0.5",
    scrollBarColor: "sbc-neutral-sf-03"
  }, f = {
    display: "block",
    width: "w-full",
    textOverflow: "truncate",
    color: "c-neutral-sf-00",
    fontSize: "fs-2",
    lineClamp: "lc-none",
    lineHeight: "lh-120%",
    textWrap: "text-wrap",
    wordBreak: "break-words"
  };
  return /* @__PURE__ */ n(x, { group: "group", cursor: "cs-help", position: "relative", children: [
    /* @__PURE__ */ t(y, { ...r({ size: p[c] }, a || {}) }),
    /* @__PURE__ */ t(
      h,
      {
        noDefaultStyle: !0,
        display: "hidden",
        width: "w-1",
        height: "h-1",
        position: "absolute",
        transformRotate: "rot-45",
        className: `${s[o]}`,
        color: e || "c-neutral-obj",
        groupHover: {
          display: "group-hover:block"
        }
      }
    ),
    /* @__PURE__ */ t(
      u,
      {
        noDefaultStyle: !0,
        ...r(b, i || {}),
        className: `${d[o]}`,
        groupHover: {
          display: "group-hover:block"
        },
        children: typeof l == "string" ? /* @__PURE__ */ t(
          g,
          {
            noDefaultStyle: !0,
            ...r(f, m || {}),
            children: l
          }
        ) : l
      }
    )
  ] });
};
export {
  M as default
};
