// @flowwwkr/design-system-tailwind@0.12.12/dist/tailwind/config.js
import { spacingTheme as e, sizingTheme as f, zIndexTheme as c, positioningTheme as a, gapTheme as r } from "./themes/index.js";
import t from "./plugins/index.js";
const s = {
  content: [
    "./node_modules/@flowwwkr/design-system-tailwind/dist/components/**/*.{js,ts,jsx,tsx,mdx}"
  ],
  theme: {
    screens: {
      xd: { max: "1920px" },
      // => extra large desktop @media (max-width: 1920px) { ... }
      ld: { max: "1440px" },
      // => large desktop @media (max-width: 1440px) { ... }
      dt: { max: "1280px" },
      // => desktop @media (max-width: 1280px) { ... }
      lt: { max: "1024px" },
      // => laptop @media (max-width: 1024px) { ... }
      tb: { max: "768px" },
      // => tablet @media (max-width: 640px) { ... }
      mb: { max: "640px" },
      // => mobile @media (max-width: 639px) { ... }
      sb: { max: "480px" },
      // => small mobile @media (max-width: 479px) { ... }
      xs: { max: "320px" },
      // => extra small @media (max-width: 319px) { ... }
      ls: { raw: "(orientation: landscape)" },
      // => landscape orientation
      pt: { raw: "(orientation: portrait)" }
      // => portrait orientation
    },
    colors: {
      "red-50": {
        light: {
          surface: "#fee2e2",
          object: "#460909"
        },
        dark: {
          surface: "#460909",
          object: "#fef2f2"
        }
      },
      "red-100": {
        light: {
          surface: "#fee2e2",
          object: "#460909"
        },
        dark: {
          surface: "#811b1b",
          object: "#fef2f2"
        }
      },
      "red-200": {
        light: {
          surface: "#ffc9c9",
          object: "#460909"
        },
        dark: {
          surface: "#9b1919",
          object: "#fef2f2"
        }
      },
      "red-300": {
        light: {
          surface: "#fea3a3",
          object: "#460909"
        },
        dark: {
          surface: "#bc1919",
          object: "#fef2f2"
        }
      },
      "red-400": {
        light: {
          surface: "#fa6f6f",
          object: "#460909"
        },
        dark: {
          surface: "#e23636",
          object: "#fef2f2"
        }
      },
      "red-500": {
        light: {
          surface: "#f24141",
          object: "#fef2f2"
        },
        dark: {
          surface: "#f24141",
          object: "#fef2f2"
        }
      },
      "red-600": {
        light: {
          surface: "#e23636",
          object: "#fef2f2"
        },
        dark: {
          surface: "#fa6f6f",
          object: "#460909"
        }
      },
      "red-700": {
        light: {
          surface: "#bc1919",
          object: "#fef2f2"
        },
        dark: {
          surface: "#fea3a3",
          object: "#460909"
        }
      },
      "red-800": {
        light: {
          surface: "#9b1919",
          object: "#fef2f2"
        },
        dark: {
          surface: "#ffc9c9",
          object: "#460909"
        }
      },
      "red-900": {
        light: {
          surface: "#811b1b",
          object: "#fef2f2"
        },
        dark: {
          surface: "#fee2e2",
          object: "#460909"
        }
      },
      "red-950": {
        light: {
          surface: "#460909",
          object: "#fef2f2"
        },
        dark: {
          surface: "#fee2e2",
          object: "#460909"
        }
      },
      "orange-50": {
        light: {
          surface: "#fff8ec",
          object: "#461704"
        },
        dark: {
          surface: "#461704",
          object: "#fff8ec"
        }
      },
      "orange-100": {
        light: {
          surface: "#fff0d3",
          object: "#461704"
        },
        dark: {
          surface: "#82330c",
          object: "#fff8ec"
        }
      },
      "orange-200": {
        light: {
          surface: "#ffdea5",
          object: "#461704"
        },
        dark: {
          surface: "#a13b0b",
          object: "#fff8ec"
        }
      },
      "orange-300": {
        light: {
          surface: "#ffc56d",
          object: "#461704"
        },
        dark: {
          surface: "#cc4c02",
          object: "#fff8ec"
        }
      },
      "orange-400": {
        light: {
          surface: "#ffa032",
          object: "#461704"
        },
        dark: {
          surface: "#ff6a00",
          object: "#fff8ec"
        }
      },
      "orange-500": {
        light: {
          surface: "#ff830a",
          object: "#fff8ec"
        },
        dark: {
          surface: "#ff830a",
          object: "#fff8ec"
        }
      },
      "orange-600": {
        light: {
          surface: "#ff6a00",
          object: "#fff8ec"
        },
        dark: {
          surface: "#ffa032",
          object: "#461704"
        }
      },
      "orange-700": {
        light: {
          surface: "#cc4c02",
          object: "#fff8ec"
        },
        dark: {
          surface: "#ffc56d",
          object: "#461704"
        }
      },
      "orange-800": {
        light: {
          surface: "#a13b0b",
          object: "#fff8ec"
        },
        dark: {
          surface: "#ffdea5",
          object: "#461704"
        }
      },
      "orange-900": {
        light: {
          surface: "#82330c",
          object: "#fff8ec"
        },
        dark: {
          surface: "#fff0d3",
          object: "#461704"
        }
      },
      "orange-950": {
        light: {
          surface: "#461704",
          object: "#fff8ec"
        },
        dark: {
          surface: "#fff8ec",
          object: "#461704"
        }
      },
      "gold-50": {
        light: {
          surface: "#fdfced",
          object: "#3e220a"
        },
        dark: {
          surface: "#3e220a",
          object: "#fdfced"
        }
      },
      "gold-100": {
        light: {
          surface: "#f8f4cd",
          object: "#3e220a"
        },
        dark: {
          surface: "#6c411b",
          object: "#fdfced"
        }
      },
      "gold-200": {
        light: {
          surface: "#f0e897",
          object: "#3e220a"
        },
        dark: {
          surface: "#83501d",
          object: "#fdfced"
        }
      },
      "gold-300": {
        light: {
          surface: "#e8d961",
          object: "#3e220a"
        },
        dark: {
          surface: "#a1651c",
          object: "#fdfced"
        }
      },
      "gold-400": {
        light: {
          surface: "#e2c636",
          object: "#3e220a"
        },
        dark: {
          surface: "#c1881e",
          object: "#fdfced"
        }
      },
      "gold-500": {
        light: {
          surface: "#dbac25",
          object: "#fdfced"
        },
        dark: {
          surface: "#dbac25",
          object: "#fdfced"
        }
      },
      "gold-600": {
        light: {
          surface: "#c1881e",
          object: "#fdfced"
        },
        dark: {
          surface: "#e2c636",
          object: "#3e220a"
        }
      },
      "gold-700": {
        light: {
          surface: "#a1651c",
          object: "#fdfced"
        },
        dark: {
          surface: "#e8d961",
          object: "#3e220a"
        }
      },
      "gold-800": {
        light: {
          surface: "#83501d",
          object: "#fdfced"
        },
        dark: {
          surface: "#f0e897",
          object: "#3e220a"
        }
      },
      "gold-900": {
        light: {
          surface: "#6c411b",
          object: "#fdfced"
        },
        dark: {
          surface: "#f8f4cd",
          object: "#3e220a"
        }
      },
      "gold-950": {
        light: {
          surface: "#3e220a",
          object: "#fdfced"
        },
        dark: {
          surface: "#fdfced",
          object: "#3e220a"
        }
      },
      "yellow-50": {
        light: {
          surface: "#fdffe7",
          object: "#442a04"
        },
        dark: {
          surface: "#442a04",
          object: "#fdffe7"
        }
      },
      "yellow-100": {
        light: {
          surface: "#f9ffc1",
          object: "#442a04"
        },
        dark: {
          surface: "#744f0f",
          object: "#fdffe7"
        }
      },
      "yellow-200": {
        light: {
          surface: "#f8ff86",
          object: "#442a04"
        },
        dark: {
          surface: "#89610a",
          object: "#fdffe7"
        }
      },
      "yellow-300": {
        light: {
          surface: "#fbff41",
          object: "#442a04"
        },
        dark: {
          surface: "#a67d02",
          object: "#fdffe7"
        }
      },
      "yellow-400": {
        light: {
          surface: "#fff80d",
          object: "#442a04"
        },
        dark: {
          surface: "#d1ae00",
          object: "#fdffe7"
        }
      },
      "yellow-500": {
        light: {
          surface: "#ffea00",
          object: "#fdffe7"
        },
        dark: {
          surface: "#ffea00",
          object: "#fdffe7"
        }
      },
      "yellow-600": {
        light: {
          surface: "#d1ae00",
          object: "#fdffe7"
        },
        dark: {
          surface: "#fff80d",
          object: "#442a04"
        }
      },
      "yellow-700": {
        light: {
          surface: "#a67d02",
          object: "#fdffe7"
        },
        dark: {
          surface: "#fbff41",
          object: "#442a04"
        }
      },
      "yellow-800": {
        light: {
          surface: "#89610a",
          object: "#fdffe7"
        },
        dark: {
          surface: "#f8ff86",
          object: "#442a04"
        }
      },
      "yellow-900": {
        light: {
          surface: "#744f0f",
          object: "#fdffe7"
        },
        dark: {
          surface: "#f9ffc1",
          object: "#442a04"
        }
      },
      "yellow-950": {
        light: {
          surface: "#442a04",
          object: "#fdffe7"
        },
        dark: {
          surface: "#fdffe7",
          object: "#442a04"
        }
      },
      "lime-50": {
        light: {
          surface: "#fbffe4",
          object: "#213400"
        },
        dark: {
          surface: "#213400",
          object: "#fbffe4"
        }
      },
      "lime-100": {
        light: {
          surface: "#f6ffc4",
          object: "#213400"
        },
        dark: {
          surface: "#415c0b",
          object: "#fbffe4"
        }
      },
      "lime-200": {
        light: {
          surface: "#ebff90",
          object: "#213400"
        },
        dark: {
          surface: "#4d6d07",
          object: "#fbffe4"
        }
      },
      "lime-300": {
        light: {
          surface: "#d9ff50",
          object: "#213400"
        },
        dark: {
          surface: "#618b00",
          object: "#fbffe4"
        }
      },
      "lime-400": {
        light: {
          surface: "#bfff00",
          object: "#213400"
        },
        dark: {
          surface: "#80b800",
          object: "#fbffe4"
        }
      },
      "lime-500": {
        light: {
          surface: "#a6e600",
          object: "#fbffe4"
        },
        dark: {
          surface: "#a6e600",
          object: "#fbffe4"
        }
      },
      "lime-600": {
        light: {
          surface: "#80b800",
          object: "#fbffe4"
        },
        dark: {
          surface: "#bfff00",
          object: "#213400"
        }
      },
      "lime-700": {
        light: {
          surface: "#618b00",
          object: "#fbffe4"
        },
        dark: {
          surface: "#d9ff50",
          object: "#213400"
        }
      },
      "lime-800": {
        light: {
          surface: "#4d6d07",
          object: "#fbffe4"
        },
        dark: {
          surface: "#ebff90",
          object: "#213400"
        }
      },
      "lime-900": {
        light: {
          surface: "#415c0b",
          object: "#fbffe4"
        },
        dark: {
          surface: "#f6ffc4",
          object: "#213400"
        }
      },
      "lime-950": {
        light: {
          surface: "#213400",
          object: "#fbffe4"
        },
        dark: {
          surface: "#fbffe4",
          object: "#213400"
        }
      },
      "green-50": {
        light: {
          surface: "#f0fee7",
          object: "#0f2d06"
        },
        dark: {
          surface: "#0f2d06",
          object: "#f0fee7"
        }
      },
      "green-100": {
        light: {
          surface: "#dffbcc",
          object: "#0f2d06"
        },
        dark: {
          surface: "#255215",
          object: "#f0fee7"
        }
      },
      "green-200": {
        light: {
          surface: "#bff79f",
          object: "#0f2d06"
        },
        dark: {
          surface: "#296014",
          object: "#f0fee7"
        }
      },
      "green-300": {
        light: {
          surface: "#96ef67",
          object: "#0f2d06"
        },
        dark: {
          surface: "#2f7a11",
          object: "#f0fee7"
        }
      },
      "green-400": {
        light: {
          surface: "#70e236",
          object: "#0f2d06"
        },
        dark: {
          surface: "#3ba010",
          object: "#f0fee7"
        }
      },
      "green-500": {
        light: {
          surface: "#52c81a",
          object: "#f0fee7"
        },
        dark: {
          surface: "#52c81a",
          object: "#f0fee7"
        }
      },
      "green-600": {
        light: {
          surface: "#3ba010",
          object: "#f0fee7"
        },
        dark: {
          surface: "#70e236",
          object: "#0f2d06"
        }
      },
      "green-700": {
        light: {
          surface: "#2f7a11",
          object: "#f0fee7"
        },
        dark: {
          surface: "#96ef67",
          object: "#0f2d06"
        }
      },
      "green-800": {
        light: {
          surface: "#296014",
          object: "#f0fee7"
        },
        dark: {
          surface: "#bff79f",
          object: "#0f2d06"
        }
      },
      "green-900": {
        light: {
          surface: "#255215",
          object: "#f0fee7"
        },
        dark: {
          surface: "#dffbcc",
          object: "#0f2d06"
        }
      },
      "green-950": {
        light: {
          surface: "#0f2d06",
          object: "#f0fee7"
        },
        dark: {
          surface: "#f0fee7",
          object: "#0f2d06"
        }
      },
      "emerald-50": {
        light: {
          surface: "#effef6",
          object: "#03301c"
        },
        dark: {
          surface: "#03301c",
          object: "#effef6"
        }
      },
      "emerald-100": {
        light: {
          surface: "#dbfdec",
          object: "#03301c"
        },
        dark: {
          surface: "#115635",
          object: "#effef6"
        }
      },
      "emerald-200": {
        light: {
          surface: "#b9f9d8",
          object: "#03301c"
        },
        dark: {
          surface: "#13683e",
          object: "#effef6"
        }
      },
      "emerald-300": {
        light: {
          surface: "#82f3ba",
          object: "#03301c"
        },
        dark: {
          surface: "#11844c",
          object: "#effef6"
        }
      },
      "emerald-400": {
        light: {
          surface: "#36e28c",
          object: "#03301c"
        },
        dark: {
          surface: "#10a95d",
          object: "#effef6"
        }
      },
      "emerald-500": {
        light: {
          surface: "#1bcc74",
          object: "#effef6"
        },
        dark: {
          surface: "#1bcc74",
          object: "#effef6"
        }
      },
      "emerald-600": {
        light: {
          surface: "#10a95d",
          object: "#effef6"
        },
        dark: {
          surface: "#36e28c",
          object: "#03301c"
        }
      },
      "emerald-700": {
        light: {
          surface: "#11844c",
          object: "#effef6"
        },
        dark: {
          surface: "#82f3ba",
          object: "#03301c"
        }
      },
      "emerald-800": {
        light: {
          surface: "#13683e",
          object: "#effef6"
        },
        dark: {
          surface: "#b9f9d8",
          object: "#03301c"
        }
      },
      "emerald-900": {
        light: {
          surface: "#115635",
          object: "#effef6"
        },
        dark: {
          surface: "#dbfdec",
          object: "#03301c"
        }
      },
      "emerald-950": {
        light: {
          surface: "#03301c",
          object: "#effef6"
        },
        dark: {
          surface: "#effef6",
          object: "#03301c"
        }
      },
      "turquoise-50": {
        light: {
          surface: "#effefb",
          object: "#013132"
        },
        dark: {
          surface: "#013132",
          object: "#effefb"
        }
      },
      "turquoise-100": {
        light: {
          surface: "#c9fef5",
          object: "#013132"
        },
        dark: {
          surface: "#0f5250",
          object: "#effefb"
        }
      },
      "turquoise-200": {
        light: {
          surface: "#93fced",
          object: "#013132"
        },
        dark: {
          surface: "#0c6361",
          object: "#effefb"
        }
      },
      "turquoise-300": {
        light: {
          surface: "#55f3e1",
          object: "#013132"
        },
        dark: {
          surface: "#087d79",
          object: "#effefb"
        }
      },
      "turquoise-400": {
        light: {
          surface: "#36e2d4",
          object: "#013132"
        },
        dark: {
          surface: "#049d96",
          object: "#effefb"
        }
      },
      "turquoise-500": {
        light: {
          surface: "#09c3b7",
          object: "#effefb"
        },
        dark: {
          surface: "#09c3b7",
          object: "#effefb"
        }
      },
      "turquoise-600": {
        light: {
          surface: "#049d96",
          object: "#effefb"
        },
        dark: {
          surface: "#36e2d4",
          object: "#013132"
        }
      },
      "turquoise-700": {
        light: {
          surface: "#087d79",
          object: "#effefb"
        },
        dark: {
          surface: "#55f3e1",
          object: "#013132"
        }
      },
      "turquoise-800": {
        light: {
          surface: "#0c6361",
          object: "#effefb"
        },
        dark: {
          surface: "#93fced",
          object: "#013132"
        }
      },
      "turquoise-900": {
        light: {
          surface: "#0f5250",
          object: "#effefb"
        },
        dark: {
          surface: "#c9fef5",
          object: "#013132"
        }
      },
      "turquoise-950": {
        light: {
          surface: "#013132",
          object: "#effefb"
        },
        dark: {
          surface: "#effefb",
          object: "#013132"
        }
      },
      "azure-50": {
        light: {
          surface: "#eef9ff",
          object: "#153056"
        },
        dark: {
          surface: "#153056",
          object: "#eef9ff"
        }
      },
      "azure-100": {
        light: {
          surface: "#daf1ff",
          object: "#153056"
        },
        dark: {
          surface: "#1a4d8e",
          object: "#eef9ff"
        }
      },
      "azure-200": {
        light: {
          surface: "#bde7ff",
          object: "#153056"
        },
        dark: {
          surface: "#195ab4",
          object: "#eef9ff"
        }
      },
      "azure-300": {
        light: {
          surface: "#8fdaff",
          object: "#153056"
        },
        dark: {
          surface: "#1a76e8",
          object: "#eef9ff"
        }
      },
      "azure-400": {
        light: {
          surface: "#5bc2ff",
          object: "#153056"
        },
        dark: {
          surface: "#1e87f2",
          object: "#eef9ff"
        }
      },
      "azure-500": {
        light: {
          surface: "#34a5fd",
          object: "#eef9ff"
        },
        dark: {
          surface: "#34a5fd",
          object: "#eef9ff"
        }
      },
      "azure-600": {
        light: {
          surface: "#1e87f2",
          object: "#eef9ff"
        },
        dark: {
          surface: "#5bc2ff",
          object: "#153056"
        }
      },
      "azure-700": {
        light: {
          surface: "#1a76e8",
          object: "#eef9ff"
        },
        dark: {
          surface: "#8fdaff",
          object: "#153056"
        }
      },
      "azure-800": {
        light: {
          surface: "#195ab4",
          object: "#eef9ff"
        },
        dark: {
          surface: "#bde7ff",
          object: "#153056"
        }
      },
      "azure-900": {
        light: {
          surface: "#1a4d8e",
          object: "#eef9ff"
        },
        dark: {
          surface: "#daf1ff",
          object: "#153056"
        }
      },
      "azure-950": {
        light: {
          surface: "#153056",
          object: "#eef9ff"
        },
        dark: {
          surface: "#eef9ff",
          object: "#153056"
        }
      },
      "blue-ribbon-50": {
        light: {
          surface: "#edf7ff",
          object: "#0e265d"
        },
        dark: {
          surface: "#0e265d",
          object: "#edf7ff"
        }
      },
      "blue-ribbon-100": {
        light: {
          surface: "#d6ecff",
          object: "#0e265d"
        },
        dark: {
          surface: "#0d3c9b",
          object: "#edf7ff"
        }
      },
      "blue-ribbon-200": {
        light: {
          surface: "#b5dfff",
          object: "#0e265d"
        },
        dark: {
          surface: "#0841c5",
          object: "#edf7ff"
        }
      },
      "blue-ribbon-300": {
        light: {
          surface: "#83cdff",
          object: "#0e265d"
        },
        dark: {
          surface: "#0055ff",
          object: "#edf7ff"
        }
      },
      "blue-ribbon-400": {
        light: {
          surface: "#48b0ff",
          object: "#0e265d"
        },
        dark: {
          surface: "#066aff",
          object: "#edf7ff"
        }
      },
      "blue-ribbon-500": {
        light: {
          surface: "#1e8bff",
          object: "#edf7ff"
        },
        dark: {
          surface: "#1e8bff",
          object: "#edf7ff"
        }
      },
      "blue-ribbon-600": {
        light: {
          surface: "#066aff",
          object: "#edf7ff"
        },
        dark: {
          surface: "#48b0ff",
          object: "#0e265d"
        }
      },
      "blue-ribbon-700": {
        light: {
          surface: "#0055ff",
          object: "#edf7ff"
        },
        dark: {
          surface: "#83cdff",
          object: "#0e265d"
        }
      },
      "blue-ribbon-800": {
        light: {
          surface: "#0841c5",
          object: "#edf7ff"
        },
        dark: {
          surface: "#b5dfff",
          object: "#0e265d"
        }
      },
      "blue-ribbon-900": {
        light: {
          surface: "#0d3c9b",
          object: "#edf7ff"
        },
        dark: {
          surface: "#d6ecff",
          object: "#0e265d"
        }
      },
      "blue-ribbon-950": {
        light: {
          surface: "#0e265d",
          object: "#edf7ff"
        },
        dark: {
          surface: "#edf7ff",
          object: "#0e265d"
        }
      },
      "royal-blue-50": {
        light: {
          surface: "#f0f5fe",
          object: "#1a2451"
        },
        dark: {
          surface: "#1a2451",
          object: "#f0f5fe"
        }
      },
      "royal-blue-100": {
        light: {
          surface: "#dde8fc",
          object: "#1a2451"
        },
        dark: {
          surface: "#233785",
          object: "#f0f5fe"
        }
      },
      "royal-blue-200": {
        light: {
          surface: "#c2d7fb",
          object: "#1a2451"
        },
        dark: {
          surface: "#253ca8",
          object: "#f0f5fe"
        }
      },
      "royal-blue-300": {
        light: {
          surface: "#98bff8",
          object: "#1a2451"
        },
        dark: {
          surface: "#2647cf",
          object: "#f0f5fe"
        }
      },
      "royal-blue-400": {
        light: {
          surface: "#689df2",
          object: "#1a2451"
        },
        dark: {
          surface: "#3661e2",
          object: "#f0f5fe"
        }
      },
      "royal-blue-500": {
        light: {
          surface: "#4479ed",
          object: "#f0f5fe"
        },
        dark: {
          surface: "#4479ed",
          object: "#f0f5fe"
        }
      },
      "royal-blue-600": {
        light: {
          surface: "#3661e2",
          object: "#f0f5fe"
        },
        dark: {
          surface: "#689df2",
          object: "#1a2451"
        }
      },
      "royal-blue-700": {
        light: {
          surface: "#2647cf",
          object: "#f0f5fe"
        },
        dark: {
          surface: "#98bff8",
          object: "#1a2451"
        }
      },
      "royal-blue-800": {
        light: {
          surface: "#253ca8",
          object: "#f0f5fe"
        },
        dark: {
          surface: "#c2d7fb",
          object: "#1a2451"
        }
      },
      "royal-blue-900": {
        light: {
          surface: "#233785",
          object: "#f0f5fe"
        },
        dark: {
          surface: "#dde8fc",
          object: "#1a2451"
        }
      },
      "royal-blue-950": {
        light: {
          surface: "#1a2451",
          object: "#f0f5fe"
        },
        dark: {
          surface: "#f0f5fe",
          object: "#1a2451"
        }
      },
      "purple-50": {
        light: {
          surface: "#eef0ff",
          object: "#231b4b"
        },
        dark: {
          surface: "#231b4b",
          object: "#eef0ff"
        }
      },
      "purple-100": {
        light: {
          surface: "#e0e4ff",
          object: "#231b4b"
        },
        dark: {
          surface: "#3a2e81",
          object: "#eef0ff"
        }
      },
      "purple-200": {
        light: {
          surface: "#c7ccfe",
          object: "#231b4b"
        },
        dark: {
          surface: "#4430a3",
          object: "#eef0ff"
        }
      },
      "purple-300": {
        light: {
          surface: "#a5aafc",
          object: "#231b4b"
        },
        dark: {
          surface: "#5339c9",
          object: "#eef0ff"
        }
      },
      "purple-400": {
        light: {
          surface: "#8381f8",
          object: "#231b4b"
        },
        dark: {
          surface: "#5336e2",
          object: "#eef0ff"
        }
      },
      "purple-500": {
        light: {
          surface: "#7164f0",
          object: "#eef0ff"
        },
        dark: {
          surface: "#7164f0",
          object: "#eef0ff"
        }
      },
      "purple-600": {
        light: {
          surface: "#5336e2",
          object: "#eef0ff"
        },
        dark: {
          surface: "#8381f8",
          object: "#231b4b"
        }
      },
      "purple-700": {
        light: {
          surface: "#5339c9",
          object: "#eef0ff"
        },
        dark: {
          surface: "#a5aafc",
          object: "#231b4b"
        }
      },
      "purple-800": {
        light: {
          surface: "#4430a3",
          object: "#eef0ff"
        },
        dark: {
          surface: "#c7ccfe",
          object: "#231b4b"
        }
      },
      "purple-900": {
        light: {
          surface: "#3a2e81",
          object: "#eef0ff"
        },
        dark: {
          surface: "#e0e4ff",
          object: "#231b4b"
        }
      },
      "purple-950": {
        light: {
          surface: "#231b4b",
          object: "#eef0ff"
        },
        dark: {
          surface: "#eef0ff",
          object: "#231b4b"
        }
      },
      "violet-50": {
        light: {
          surface: "#fbf5ff",
          object: "#400b60"
        },
        dark: {
          surface: "#400b60",
          object: "#fbf5ff"
        }
      },
      "violet-100": {
        light: {
          surface: "#f4e9fe",
          object: "#400b60"
        },
        dark: {
          surface: "#5d2083",
          object: "#fbf5ff"
        }
      },
      "violet-200": {
        light: {
          surface: "#ebd7fd",
          object: "#400b60"
        },
        dark: {
          surface: "#7226a3",
          object: "#fbf5ff"
        }
      },
      "violet-300": {
        light: {
          surface: "#dcb7fb",
          object: "#400b60"
        },
        dark: {
          surface: "#8829c7",
          object: "#fbf5ff"
        }
      },
      "violet-400": {
        light: {
          surface: "#c789f7",
          object: "#400b60"
        },
        dark: {
          surface: "#9b36e2",
          object: "#fbf5ff"
        }
      },
      "violet-500": {
        light: {
          surface: "#b15cf0",
          object: "#fbf5ff"
        },
        dark: {
          surface: "#b15cf0",
          object: "#fbf5ff"
        }
      },
      "violet-600": {
        light: {
          surface: "#9b36e2",
          object: "#fbf5ff"
        },
        dark: {
          surface: "#c789f7",
          object: "#400b60"
        }
      },
      "violet-700": {
        light: {
          surface: "#8829c7",
          object: "#fbf5ff"
        },
        dark: {
          surface: "#dcb7fb",
          object: "#400b60"
        }
      },
      "violet-800": {
        light: {
          surface: "#7226a3",
          object: "#fbf5ff"
        },
        dark: {
          surface: "#ebd7fd",
          object: "#400b60"
        }
      },
      "violet-900": {
        light: {
          surface: "#5d2083",
          object: "#fbf5ff"
        },
        dark: {
          surface: "#f4e9fe",
          object: "#400b60"
        }
      },
      "violet-950": {
        light: {
          surface: "#400b60",
          object: "#fbf5ff"
        },
        dark: {
          surface: "#fbf5ff",
          object: "#400b60"
        }
      },
      "fuchsia-50": {
        light: {
          surface: "#fef5fe",
          object: "#4a0845"
        },
        dark: {
          surface: "#4a0845",
          object: "#fef5fe"
        }
      },
      "fuchsia-100": {
        light: {
          surface: "#fce9fe",
          object: "#4a0845"
        },
        dark: {
          surface: "#701f69",
          object: "#fef5fe"
        }
      },
      "fuchsia-200": {
        light: {
          surface: "#f8d3fb",
          object: "#4a0845"
        },
        dark: {
          surface: "#882082",
          object: "#fef5fe"
        }
      },
      "fuchsia-300": {
        light: {
          surface: "#f7aff8",
          object: "#4a0845"
        },
        dark: {
          surface: "#a724a2",
          object: "#fef5fe"
        }
      },
      "fuchsia-400": {
        light: {
          surface: "#f280f2",
          object: "#4a0845"
        },
        dark: {
          surface: "#c930c5",
          object: "#fef5fe"
        }
      },
      "fuchsia-500": {
        light: {
          surface: "#e236e2",
          object: "#fef5fe"
        },
        dark: {
          surface: "#e236e2",
          object: "#fef5fe"
        }
      },
      "fuchsia-600": {
        light: {
          surface: "#c930c5",
          object: "#fef5fe"
        },
        dark: {
          surface: "#f280f2",
          object: "#4a0845"
        }
      },
      "fuchsia-700": {
        light: {
          surface: "#a724a2",
          object: "#fef5fe"
        },
        dark: {
          surface: "#f7aff8",
          object: "#4a0845"
        }
      },
      "fuchsia-800": {
        light: {
          surface: "#882082",
          object: "#fef5fe"
        },
        dark: {
          surface: "#f8d3fb",
          object: "#4a0845"
        }
      },
      "fuchsia-900": {
        light: {
          surface: "#701f69",
          object: "#fef5fe"
        },
        dark: {
          surface: "#fce9fe",
          object: "#4a0845"
        }
      },
      "fuchsia-950": {
        light: {
          surface: "#4a0845",
          object: "#fef5fe"
        },
        dark: {
          surface: "#fef5fe",
          object: "#4a0845"
        }
      },
      "rose-50": {
        light: {
          surface: "#fff0fb",
          object: "#5f0036"
        },
        dark: {
          surface: "#5f0036",
          object: "#fff0fb"
        }
      },
      "rose-100": {
        light: {
          surface: "#ffe3fa",
          object: "#5f0036"
        },
        dark: {
          surface: "#980360",
          object: "#fff0fb"
        }
      },
      "rose-200": {
        light: {
          surface: "#ffc6f6",
          object: "#5f0036"
        },
        dark: {
          surface: "#b80071",
          object: "#fff0fb"
        }
      },
      "rose-300": {
        light: {
          surface: "#ff98ec",
          object: "#5f0036"
        },
        dark: {
          surface: "#df0089",
          object: "#fff0fb"
        }
      },
      "rose-400": {
        light: {
          surface: "#ff58dc",
          object: "#5f0036"
        },
        dark: {
          surface: "#ff00aa",
          object: "#fff0fb"
        }
      },
      "rose-500": {
        light: {
          surface: "#ff27c9",
          object: "#fff0fb"
        },
        dark: {
          surface: "#ff27c9",
          object: "#fff0fb"
        }
      },
      "rose-600": {
        light: {
          surface: "#ff00aa",
          object: "#fff0fb"
        },
        dark: {
          surface: "#ff58dc",
          object: "#5f0036"
        }
      },
      "rose-700": {
        light: {
          surface: "#df0089",
          object: "#fff0fb"
        },
        dark: {
          surface: "#ff98ec",
          object: "#5f0036"
        }
      },
      "rose-800": {
        light: {
          surface: "#b80071",
          object: "#fff0fb"
        },
        dark: {
          surface: "#ffc6f6",
          object: "#5f0036"
        }
      },
      "rose-900": {
        light: {
          surface: "#980360",
          object: "#fff0fb"
        },
        dark: {
          surface: "#ffe3fa",
          object: "#5f0036"
        }
      },
      "rose-950": {
        light: {
          surface: "#5f0036",
          object: "#fff0fb"
        },
        dark: {
          surface: "#fff0fb",
          object: "#5f0036"
        }
      },
      "gs-cool-50": {
        light: {
          surface: "#f6f6f7",
          object: "#303136"
        },
        dark: {
          surface: "#303136",
          object: "#f6f6f7"
        }
      },
      "gs-cool-100": {
        light: {
          surface: "#eef0f1",
          object: "#303136"
        },
        dark: {
          surface: "#53555c",
          object: "#f6f6f7"
        }
      },
      "gs-cool-200": {
        light: {
          surface: "#dfe3e6",
          object: "#303136"
        },
        dark: {
          surface: "#636670",
          object: "#f6f6f7"
        }
      },
      "gs-cool-300": {
        light: {
          surface: "#cbd0d6",
          object: "#303136"
        },
        dark: {
          surface: "#787b8a",
          object: "#f6f6f7"
        }
      },
      "gs-cool-400": {
        light: {
          surface: "#b6bbc3",
          object: "#303136"
        },
        dark: {
          surface: "#8c909f",
          object: "#f6f6f7"
        }
      },
      "gs-cool-500": {
        light: {
          surface: "#a0a5b1",
          object: "#f6f6f7"
        },
        dark: {
          surface: "#a0a5b1",
          object: "#f6f6f7"
        }
      },
      "gs-cool-600": {
        light: {
          surface: "#8c909f",
          object: "#f6f6f7"
        },
        dark: {
          surface: "#b6bbc3",
          object: "#303136"
        }
      },
      "gs-cool-700": {
        light: {
          surface: "#787b8a",
          object: "#f6f6f7"
        },
        dark: {
          surface: "#cbd0d6",
          object: "#303136"
        }
      },
      "gs-cool-800": {
        light: {
          surface: "#636670",
          object: "#f6f6f7"
        },
        dark: {
          surface: "#dfe3e6",
          object: "#303136"
        }
      },
      "gs-cool-900": {
        light: {
          surface: "#53555c",
          object: "#f6f6f7"
        },
        dark: {
          surface: "#eef0f1",
          object: "#303136"
        }
      },
      "gs-cool-950": {
        light: {
          surface: "#303136",
          object: "#f6f6f7"
        },
        dark: {
          surface: "#f6f6f7",
          object: "#303136"
        }
      },
      "gs-forest-50": {
        light: {
          surface: "#f6f7f6",
          object: "#171c17"
        },
        dark: {
          surface: "#171c17",
          object: "#f6f7f6"
        }
      },
      "gs-forest-100": {
        light: {
          surface: "#e2e6e1",
          object: "#171c17"
        },
        dark: {
          surface: "#2e342d",
          object: "#f6f7f6"
        }
      },
      "gs-forest-200": {
        light: {
          surface: "#c5cdc2",
          object: "#171c17"
        },
        dark: {
          surface: "#343c33",
          object: "#f6f7f6"
        }
      },
      "gs-forest-300": {
        light: {
          surface: "#94a38f",
          object: "#171c17"
        },
        dark: {
          surface: "#3e493c",
          object: "#f6f7f6"
        }
      },
      "gs-forest-400": {
        light: {
          surface: "#7b8b76",
          object: "#171c17"
        },
        dark: {
          surface: "#4b5948",
          object: "#f6f7f6"
        }
      },
      "gs-forest-500": {
        light: {
          surface: "#60705c",
          object: "#f6f7f6"
        },
        dark: {
          surface: "#60705c",
          object: "#f6f7f6"
        }
      },
      "gs-forest-600": {
        light: {
          surface: "#4b5948",
          object: "#f6f7f6"
        },
        dark: {
          surface: "#7b8b76",
          object: "#171c17"
        }
      },
      "gs-forest-700": {
        light: {
          surface: "#3e493c",
          object: "#f6f7f6"
        },
        dark: {
          surface: "#94a38f",
          object: "#171c17"
        }
      },
      "gs-forest-800": {
        light: {
          surface: "#343c33",
          object: "#f6f7f6"
        },
        dark: {
          surface: "#c5cdc2",
          object: "#171c17"
        }
      },
      "gs-forest-900": {
        light: {
          surface: "#2e342d",
          object: "#f6f7f6"
        },
        dark: {
          surface: "#e2e6e1",
          object: "#171c17"
        }
      },
      "gs-forest-950": {
        light: {
          surface: "#171c17",
          object: "#f6f7f6"
        },
        dark: {
          surface: "#f6f7f6",
          object: "#171c17"
        }
      },
      "gs-green-50": {
        light: {
          surface: "#f6f7f7",
          object: "#171c1b"
        },
        dark: {
          surface: "#171c1b",
          object: "#f6f7f7"
        }
      },
      "gs-green-100": {
        light: {
          surface: "#e1e6e4",
          object: "#171c1b"
        },
        dark: {
          surface: "#2d3432",
          object: "#f6f7f7"
        }
      },
      "gs-green-200": {
        light: {
          surface: "#c2cdc8",
          object: "#171c1b"
        },
        dark: {
          surface: "#333c39",
          object: "#f6f7f7"
        }
      },
      "gs-green-300": {
        light: {
          surface: "#8fa39a",
          object: "#171c1b"
        },
        dark: {
          surface: "#3c4944",
          object: "#f6f7f7"
        }
      },
      "gs-green-400": {
        light: {
          surface: "#768b82",
          object: "#171c1b"
        },
        dark: {
          surface: "#485953",
          object: "#f6f7f7"
        }
      },
      "gs-green-500": {
        light: {
          surface: "#5c7068",
          object: "#f6f7f7"
        },
        dark: {
          surface: "#5c7068",
          object: "#f6f7f7"
        }
      },
      "gs-green-600": {
        light: {
          surface: "#485953",
          object: "#f6f7f7"
        },
        dark: {
          surface: "#768b82",
          object: "#171c1b"
        }
      },
      "gs-green-700": {
        light: {
          surface: "#3c4944",
          object: "#f6f7f7"
        },
        dark: {
          surface: "#8fa39a",
          object: "#171c1b"
        }
      },
      "gs-green-800": {
        light: {
          surface: "#333c39",
          object: "#f6f7f7"
        },
        dark: {
          surface: "#c2cdc8",
          object: "#171c1b"
        }
      },
      "gs-green-900": {
        light: {
          surface: "#2d3432",
          object: "#f6f7f7"
        },
        dark: {
          surface: "#e1e6e4",
          object: "#171c1b"
        }
      },
      "gs-green-950": {
        light: {
          surface: "#171c1b",
          object: "#f6f7f7"
        },
        dark: {
          surface: "#f6f7f7",
          object: "#171c1b"
        }
      },
      "gs-pure-50": {
        light: {
          surface: "#f6f6f6",
          object: "#000000"
        },
        dark: {
          surface: "#3d3d3d",
          object: "#ffffff"
        }
      },
      "gs-pure-100": {
        light: {
          surface: "#e7e7e7",
          object: "#000000"
        },
        dark: {
          surface: "#454545",
          object: "#ffffff"
        }
      },
      "gs-pure-200": {
        light: {
          surface: "#d1d1d1",
          object: "#000000"
        },
        dark: {
          surface: "#4f4f4f",
          object: "#ffffff"
        }
      },
      "gs-pure-300": {
        light: {
          surface: "#b0b0b0",
          object: "#000000"
        },
        dark: {
          surface: "#5d5d5d",
          object: "#ffffff"
        }
      },
      "gs-pure-400": {
        light: {
          surface: "#888888",
          object: "#000000"
        },
        dark: {
          surface: "#6d6d6d",
          object: "#ffffff"
        }
      },
      "gs-pure-500": {
        light: {
          surface: "#6d6d6d",
          object: "#ffffff"
        },
        dark: {
          surface: "#6d6d6d",
          object: "#ffffff"
        }
      },
      "gs-pure-600": {
        light: {
          surface: "#5d5d5d",
          object: "#ffffff"
        },
        dark: {
          surface: "#888888",
          object: "#000000"
        }
      },
      "gs-pure-700": {
        light: {
          surface: "#4f4f4f",
          object: "#ffffff"
        },
        dark: {
          surface: "#b0b0b0",
          object: "#000000"
        }
      },
      "gs-pure-800": {
        light: {
          surface: "#454545",
          object: "#ffffff"
        },
        dark: {
          surface: "#d1d1d1",
          object: "#000000"
        }
      },
      "gs-pure-900": {
        light: {
          surface: "#3d3d3d",
          object: "#ffffff"
        },
        dark: {
          surface: "#e7e7e7",
          object: "#000000"
        }
      },
      "gs-pure-black": {
        light: {
          surface: "#000000",
          object: "#ffffff"
        },
        dark: {
          surface: "#ffffff",
          object: "#000000"
        }
      },
      "gs-pure-white": {
        light: {
          surface: "#ffffff",
          object: "#000000"
        },
        dark: {
          surface: "#000000",
          object: "#ffffff"
        }
      },
      "gs-soil-50": {
        light: {
          surface: "#f9f7f7",
          object: "#2b2424"
        },
        dark: {
          surface: "#544848",
          object: "#f9f7f7"
        }
      },
      "gs-soil-100": {
        light: {
          surface: "#f2eeee",
          object: "#2b2424"
        },
        dark: {
          surface: "#615353",
          object: "#f9f7f7"
        }
      },
      "gs-soil-200": {
        light: {
          surface: "#e7e1e1",
          object: "#2b2424"
        },
        dark: {
          surface: "#746161",
          object: "#f9f7f7"
        }
      },
      "gs-soil-300": {
        light: {
          surface: "#d6cbcb",
          object: "#2b2424"
        },
        dark: {
          surface: "#8c7676",
          object: "#f9f7f7"
        }
      },
      "gs-soil-400": {
        light: {
          surface: "#bcadad",
          object: "#2b2424"
        },
        dark: {
          surface: "#a38f8f",
          object: "#f9f7f7"
        }
      },
      "gs-soil-500": {
        light: {
          surface: "#a38f8f",
          object: "#f9f7f7"
        },
        dark: {
          surface: "#a38f8f",
          object: "#f9f7f7"
        }
      },
      "gs-soil-600": {
        light: {
          surface: "#8c7676",
          object: "#f9f7f7"
        },
        dark: {
          surface: "#bcadad",
          object: "#2b2424"
        }
      },
      "gs-soil-700": {
        light: {
          surface: "#746161",
          object: "#f9f7f7"
        },
        dark: {
          surface: "#d6cbcb",
          object: "#2b2424"
        }
      },
      "gs-soil-800": {
        light: {
          surface: "#615353",
          object: "#f9f7f7"
        },
        dark: {
          surface: "#e7e1e1",
          object: "#2b2424"
        }
      },
      "gs-soil-900": {
        light: {
          surface: "#544848",
          object: "#f9f7f7"
        },
        dark: {
          surface: "#f2eeee",
          object: "#2b2424"
        }
      },
      "gs-soil-950": {
        light: {
          surface: "#2b2424",
          object: "#f9f7f7"
        },
        dark: {
          surface: "#f9f7f7",
          object: "#2b2424"
        }
      },
      "gs-violet-50": {
        light: {
          surface: "#f9f8fa",
          object: "#3a3540"
        },
        dark: {
          surface: "#595260",
          object: "#f9f8fa"
        }
      },
      "gs-violet-100": {
        light: {
          surface: "#f3f2f5",
          object: "#3a3540"
        },
        dark: {
          surface: "#6c6374",
          object: "#f9f8fa"
        }
      },
      "gs-violet-200": {
        light: {
          surface: "#e9e8ec",
          object: "#3a3540"
        },
        dark: {
          surface: "#81768b",
          object: "#f9f8fa"
        }
      },
      "gs-violet-300": {
        light: {
          surface: "#d8d5dd",
          object: "#3a3540"
        },
        dark: {
          surface: "#998fa3",
          object: "#f9f8fa"
        }
      },
      "gs-violet-400": {
        light: {
          surface: "#c1bcc9",
          object: "#3a3540"
        },
        dark: {
          surface: "#a7a0b2",
          object: "#f9f8fa"
        }
      },
      "gs-violet-500": {
        light: {
          surface: "#a7a0b2",
          object: "#f9f8fa"
        },
        dark: {
          surface: "#a7a0b2",
          object: "#f9f8fa"
        }
      },
      "gs-violet-600": {
        light: {
          surface: "#998fa3",
          object: "#f9f8fa"
        },
        dark: {
          surface: "#c1bcc9",
          object: "#3a3540"
        }
      },
      "gs-violet-700": {
        light: {
          surface: "#81768b",
          object: "#f9f8fa"
        },
        dark: {
          surface: "#d8d5dd",
          object: "#3a3540"
        }
      },
      "gs-violet-800": {
        light: {
          surface: "#6c6374",
          object: "#f9f8fa"
        },
        dark: {
          surface: "#e9e8ec",
          object: "#3a3540"
        }
      },
      "gs-violet-900": {
        light: {
          surface: "#595260",
          object: "#f9f8fa"
        },
        dark: {
          surface: "#f3f2f5",
          object: "#3a3540"
        }
      },
      "gs-violet-950": {
        light: {
          surface: "#3a3540",
          object: "#f9f8fa"
        },
        dark: {
          surface: "#f9f8fa",
          object: "#3a3540"
        }
      },
      "gs-warm-50": {
        light: {
          surface: "#f8f7f8",
          object: "#30272d"
        },
        dark: {
          surface: "#54474f",
          object: "#f8f7f8"
        }
      },
      "gs-warm-100": {
        light: {
          surface: "#f3f0f3",
          object: "#30272d"
        },
        dark: {
          surface: "#62525c",
          object: "#f8f7f8"
        }
      },
      "gs-warm-200": {
        light: {
          surface: "#e8e2e7",
          object: "#30272d"
        },
        dark: {
          surface: "#75616e",
          object: "#f8f7f8"
        }
      },
      "gs-warm-300": {
        light: {
          surface: "#d5ccd3",
          object: "#30272d"
        },
        dark: {
          surface: "#8c7686",
          object: "#f8f7f8"
        }
      },
      "gs-warm-400": {
        light: {
          surface: "#bbabb7",
          object: "#30272d"
        },
        dark: {
          surface: "#a38f9e",
          object: "#f8f7f8"
        }
      },
      "gs-warm-500": {
        light: {
          surface: "#a38f9e",
          object: "#f8f7f8"
        },
        dark: {
          surface: "#a38f9e",
          object: "#f8f7f8"
        }
      },
      "gs-warm-600": {
        light: {
          surface: "#8c7686",
          object: "#f8f7f8"
        },
        dark: {
          surface: "#bbabb7",
          object: "#30272d"
        }
      },
      "gs-warm-700": {
        light: {
          surface: "#75616e",
          object: "#f8f7f8"
        },
        dark: {
          surface: "#d5ccd3",
          object: "#30272d"
        }
      },
      "gs-warm-800": {
        light: {
          surface: "#62525c",
          object: "#f8f7f8"
        },
        dark: {
          surface: "#e8e2e7",
          object: "#30272d"
        }
      },
      "gs-warm-900": {
        light: {
          surface: "#54474f",
          object: "#f8f7f8"
        },
        dark: {
          surface: "#f3f0f3",
          object: "#30272d"
        }
      },
      "gs-warm-950": {
        light: {
          surface: "#30272d",
          object: "#f8f7f8"
        },
        dark: {
          surface: "#f8f7f8",
          object: "#30272d"
        }
      },
      "gs-wood-50": {
        light: {
          surface: "#f7f6f5",
          object: "#282422"
        },
        dark: {
          surface: "#4c4642",
          object: "#f7f6f5"
        }
      },
      "gs-wood-100": {
        light: {
          surface: "#f2eeee",
          object: "#282422"
        },
        dark: {
          surface: "#5d564f",
          object: "#f7f6f5"
        }
      },
      "gs-wood-200": {
        light: {
          surface: "#d8d6d0",
          object: "#282422"
        },
        dark: {
          surface: "#70675e",
          object: "#f7f6f5"
        }
      },
      "gs-wood-300": {
        light: {
          surface: "#bfbcb2",
          object: "#282422"
        },
        dark: {
          surface: "#867c70",
          object: "#f7f6f5"
        }
      },
      "gs-wood-400": {
        light: {
          surface: "#a39d8f",
          object: "#282422"
        },
        dark: {
          surface: "#938b7c",
          object: "#f7f6f5"
        }
      },
      "gs-wood-500": {
        light: {
          surface: "#938b7c",
          object: "#f7f6f5"
        },
        dark: {
          surface: "#938b7c",
          object: "#f7f6f5"
        }
      },
      "gs-wood-600": {
        light: {
          surface: "#867c70",
          object: "#f7f6f5"
        },
        dark: {
          surface: "#a39d8f",
          object: "#282422"
        }
      },
      "gs-wood-700": {
        light: {
          surface: "#70675e",
          object: "#f7f6f5"
        },
        dark: {
          surface: "#bfbcb2",
          object: "#282422"
        }
      },
      "gs-wood-800": {
        light: {
          surface: "#5d564f",
          object: "#f7f6f5"
        },
        dark: {
          surface: "#d8d6d0",
          object: "#282422"
        }
      },
      "gs-wood-900": {
        light: {
          surface: "#4c4642",
          object: "#f7f6f5"
        },
        dark: {
          surface: "#f2eeee",
          object: "#282422"
        }
      },
      "gs-wood-950": {
        light: {
          surface: "#282422",
          object: "#f7f6f5"
        },
        dark: {
          surface: "#f7f6f5",
          object: "#282422"
        }
      },
      inherit: {
        light: {
          surface: "inherit",
          object: "#0000"
        },
        dark: {
          surface: "inherit",
          object: "#fff0"
        }
      },
      transparent: {
        light: {
          surface: "#fff0",
          object: "#000"
        },
        dark: {
          surface: "#0000",
          object: "#fff"
        }
      }
    },
    spacing: e,
    width: f,
    height: f,
    zIndex: c,
    flexBasis: a,
    gap: r
  },
  corePlugins: {
    alignContent: !1,
    alignItems: !1,
    borderStyle: !1,
    borderWidth: !1,
    borderColor: !1,
    borderRadius: !1,
    borderSpacing: !1,
    boxShadow: !1,
    backgroundColor: !1,
    textColor: !1,
    flexBasis: !1,
    flexGrow: !1,
    flexShrink: !1,
    height: !1,
    inset: !1,
    justifyContent: !1,
    minHeight: !1,
    maxHeight: !1,
    minWidth: !1,
    maxWidth: !1,
    objectPosition: !1,
    objectFit: !1,
    opacity: !1,
    overflow: !1,
    size: !1,
    width: !1,
    aspectRatio: !1,
    gradientColorStops: !1,
    backgroundOpacity: !1,
    gap: !1,
    fontFamily: !1,
    fontSize: !1,
    fontWeight: !1,
    userSelect: !1,
    pointerEvents: !1,
    overscrollBehavior: !1
  },
  plugins: [t],
  darkMode: ["selector", '[data-theme="dark"]']
};
export {
  s as default
};
